<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2026-01-27" />

<title>RNA_DGE_Comp_Species_Ensemble</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CrossSpecies_CM_Diff_RNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">RNA_DGE_Comp_Species_Ensemble</h1>
<h4 class="date">2026-01-27</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-02-18
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CrossSpecies_CM_Diff_RNA/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20251129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20251129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20251129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20251129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomJohnDHurleyCrossSpeciesCMDiffRNAtree888d26224324067b3022ba56e2dd2e45151cb17ctargetblank888d262a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/tree/888d26224324067b3022ba56e2dd2e45151cb17c" target="_blank">888d262</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomJohnDHurleyCrossSpeciesCMDiffRNAtree888d26224324067b3022ba56e2dd2e45151cb17ctargetblank888d262a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/tree/888d26224324067b3022ba56e2dd2e45151cb17c" target="_blank">888d262</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/RNA_DGE_Comp_Species_Ensemble.Rmd</code>) and HTML
(<code>docs/RNA_DGE_Comp_Species_Ensemble.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/888d26224324067b3022ba56e2dd2e45151cb17c/analysis/RNA_DGE_Comp_Species_Ensemble.Rmd" target="_blank">888d262</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-02-18
</td>
<td>
Block Naming Duplication
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/fce84157ecc5af29240bb5d56b620c78c9fba7c5/analysis/RNA_DGE_Comp_Species_Ensemble.Rmd" target="_blank">fce8415</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-02-18
</td>
<td>
Typo
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/b8dfde38c2c2747806f735301e69936058b2589f/analysis/RNA_DGE_Comp_Species_Ensemble.Rmd" target="_blank">b8dfde3</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-02-18
</td>
<td>
MovedClustersToCormotif
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/5d20f97af4cc10432ead420062264c76e3f4782a/analysis/RNA_DGE_Comp_Species_Ensemble.Rmd" target="_blank">5d20f97</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-02-18
</td>
<td>
WebsiteUpdates_DGE_Species
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/c85837b9cf4737e7032493f7577a8de43816a5ca/analysis/RNA_DGE_Comp_Species_Ensemble.Rmd" target="_blank">c85837b</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-02-18
</td>
<td>
SiteUpDate_CorrelationHeatMap
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/b0ab413db9b92315393162a697a43c146d357b7d/docs/RNA_DGE_Comp_Species_Ensemble.html" target="_blank">b0ab413</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/c45c0b644d53d1faa9a8fc2701d8a6a355ccf484/analysis/RNA_DGE_Comp_Species_Ensemble.Rmd" target="_blank">c45c0b6</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Erorr in workflowr publish
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/4174688a1f348aa06587f90f90263933e575bf62/analysis/RNA_DGE_Comp_Species_Ensemble.Rmd" target="_blank">4174688</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Website updates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/e801b171ce92e740cee75cd67c0a8908f104bb77/analysis/RNA_DGE_Comp_Species_Ensemble.Rmd" target="_blank">e801b17</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Duplicate Code block titles
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/92e63cc09a36a8b5046ddf1261afc4fd87567a8b/analysis/RNA_DGE_Comp_Species_Ensemble.Rmd" target="_blank">92e63cc</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
DGE for website fixes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/6c3cce3528b84cd2a91f8252b3f40ed4706edfbe/analysis/RNA_DGE_Comp_Species_Ensemble.Rmd" target="_blank">6c3cce3</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Cormotif Species
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/085c1db0f6d1cd1623c082c15b91e46e699b4cc6/analysis/RNA_DGE_Comp_Species_Ensemble.Rmd" target="_blank">085c1db</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Finalizing CorHeatMap
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>generate_volcano_plot &lt;- function(toptable, title) {
  
  # #check for entrezid 
  # if(!&quot;Entrez_ID&quot; %in% colnames(toptable)) stop(&quot;Entrez_ID col not present&quot;)
  # 
  #make significance labels
  
  toptable &lt;- toptable %&gt;% 
    mutate(Significance = case_when(
      logFC &gt; 0 &amp; adj.P.Val &lt; 0.05 ~ &quot;Upregulated&quot;,
      logFC &lt; 0 &amp; adj.P.Val &lt; 0.05 ~ &quot;Downregulated&quot;,
      TRUE  ~ &quot;Not Significant&quot;
    ))
  
  #factor significance
  toptable$Significance &lt;- factor(
    toptable$Significance, 
    levels = c(&quot;Upregulated&quot;, 
               &quot;Not Significant&quot;, 
               &quot;Downregulated&quot;)
  )
  
  #count genes in each category
  upgenes &lt;- sum(toptable$Significance == &quot;Upregulated&quot;)
  nsgenes &lt;- sum(toptable$Significance == &quot;Not Significant&quot;)
  downgenes &lt;- sum(toptable$Significance == &quot;Downregulated&quot;)
  
  
  #labels for legend
  legend_lab &lt;- c(
    paste0(&quot;Upregulated: &quot;, upgenes),
    paste0(&quot;Not Significant: &quot;, nsgenes),
    paste0(&quot;Downregulated: &quot;, downgenes)
  )
  
  #colors 
  color_map &lt;- c(&quot;Upregulated&quot; = &quot;blue&quot;,
                 &quot;Not Significant&quot; = &quot;grey30&quot;,
                 &quot;Downregulated&quot; = &quot;red&quot;)
  
  #generate volcano plots
  p &lt;- ggplot(toptable, aes(x = logFC, 
                            y = -log10(P.Value),
                            color = Significance)) +
    geom_point_rast(alpha = 0.8, size = 3) +
    scale_color_manual(values = color_map, 
                       labels = legend_lab,
                       breaks = c(&quot;Upregulated&quot;,
                                  &quot;Not Significant&quot;, 
                                  &quot;Downregulated&quot;))  +
    xlim(-20,20) +
    labs(title = title, 
         x = expression(&quot;log&quot;[2]*&quot;FC&quot;),
         y = expression(&quot;-log&quot;[10]*&quot;p-value&quot;)) +
    # theme_custom() +
    theme(legend.position = &quot;right&quot;)
  
  return(p)
}</code></pre>
<pre class="r"><code>plot_cluster_fc &lt;- function(gene_vector, cluster_name, top_table) {
  
  # Filter TopTable for the genes in this cluster
  df &lt;- top_table %&gt;% dplyr::filter(Gene %in% gene_vector) %&gt;%
    mutate(Comparisons = factor(Comparisons, levels = c(
      &quot;HC_D0&quot;,&quot;HC_D2&quot;,&quot;HC_D5&quot;,&quot;HC_D15&quot;,&quot;HC_D30&quot;,&quot;All_Genes&quot;,&quot;All_DEGs&quot;
    )))
  
  # Optional: print dimensions and unique genes
  cat(cluster_name, &quot;:&quot;, dim(df)[1], &quot;rows,&quot;, length(unique(df$Gene)), &quot;unique genes\n&quot;)
  
  # Create boxplot
  p &lt;- ggplot(df, aes(x = Comparisons, y = logFC)) +
    geom_boxplot(aes(fill = Comparisons)) +
    guides(fill = guide_legend(title = &quot;Comparisons&quot;)) +
    theme_bw() +
    xlab(&quot; &quot;) +
    ylab(&quot;logFC&quot;) +
    ggtitle(paste0(cluster_name, &quot; RNA FC Comparisons&quot;)) +
    theme(
      plot.title = element_text(size = rel(1.5), hjust = 0.5),
      axis.title = element_text(size = 15, colour = &quot;black&quot;),
      strip.background = element_rect(fill = &quot;#CAD899&quot;),
      axis.text.x = element_text(size = 8, colour = &quot;white&quot;, angle = 15),
      strip.text.x = element_text(size = 12, colour = &quot;black&quot;, face = &quot;bold&quot;)
    )
  
  return(p)
}</code></pre>
<pre class="r"><code>plot_cluster_fc_sig &lt;- function(gene_vector, cluster_name, top_table) {

  df &lt;- top_table %&gt;%
    dplyr::filter(Gene %in% gene_vector) %&gt;%
    mutate(Comparisons = factor(Comparisons, levels = c(
      &quot;HC_D0&quot;,&quot;HC_D2&quot;,&quot;HC_D5&quot;,&quot;HC_D15&quot;,&quot;HC_D30&quot;
    )))

  cat(cluster_name, &quot;:&quot;, nrow(df), &quot;rows,&quot;,
      length(unique(df$Gene)), &quot;unique genes\n&quot;)

  # Global test across comparisons
  kw &lt;- kruskal.test(logFC ~ Comparisons, data = df)
  
  pw_kw &lt;-pairwise.wilcox.test(
    df$logFC,
    df$Comparisons,
    p.adjust.method = &quot;BH&quot;)
  
  print(pw_kw)

  p &lt;- ggplot(df, aes(x = Comparisons, y = logFC)) +
    geom_boxplot(aes(fill = Comparisons)) +
    theme_bw() +
    xlab(&quot; &quot;) +
    ylab(&quot;logFC&quot;) +
    ggtitle(
      paste0(
        cluster_name,
        &quot; RNA FC Comparisons\nKruskal–Wallis p = &quot;,
        signif(kw$p.value, 3)
      )
    ) +
    theme(
      plot.title = element_text(size = rel(1.3), hjust = 0.5),
      axis.title = element_text(size = 15),
      axis.text.x = element_text(size = 8, angle = 15)
    )

  return(p)


}</code></pre>
<pre class="r"><code>plot_cluster_absfc &lt;- function(gene_vector, cluster_name, top_table) {
  
  # Filter TopTable for the genes in this cluster
  df &lt;- top_table %&gt;% dplyr::filter(Gene %in% gene_vector) %&gt;%
    mutate(Comparisons = factor(Comparisons, levels = c(
      &quot;HC_D0&quot;,&quot;HC_D2&quot;,&quot;HC_D5&quot;,&quot;HC_D15&quot;,&quot;HC_D30&quot;
    )))
  
  # Optional: print dimensions and unique genes
  cat(cluster_name, &quot;:&quot;, dim(df)[1], &quot;rows,&quot;, length(unique(df$Gene)), &quot;unique genes\n&quot;)
  
  # Create boxplot
  p &lt;- ggplot(df, aes(x = Comparisons, y = AbsFC)) +
    geom_boxplot(aes(fill = Comparisons)) +
    guides(fill = guide_legend(title = &quot;Comparisons&quot;)) +
    theme_bw() +
    xlab(&quot; &quot;) +
    ylab(&quot;AbsFC&quot;) +
    ggtitle(paste0(cluster_name, &quot; RNA AbsFC Comparisons&quot;)) +
    theme(
      plot.title = element_text(size = rel(1.5), hjust = 0.5),
      axis.title = element_text(size = 15, colour = &quot;black&quot;),
      strip.background = element_rect(fill = &quot;#CAD899&quot;),
      axis.text.x = element_text(size = 8, colour = &quot;white&quot;, angle = 15),
      strip.text.x = element_text(size = 12, colour = &quot;black&quot;, face = &quot;bold&quot;)
    )
  
  return(p)
}</code></pre>
<pre class="r"><code>plot_cluster_absfc_sig &lt;- function(gene_vector, cluster_name, top_table) {

  df &lt;- top_table %&gt;%
    dplyr::filter(Gene %in% gene_vector) %&gt;%
    mutate(Comparisons = factor(Comparisons, levels = c(
      &quot;HC_D0&quot;,&quot;HC_D2&quot;,&quot;HC_D5&quot;,&quot;HC_D15&quot;,&quot;HC_D30&quot;
    )))

  cat(cluster_name, &quot;:&quot;, nrow(df), &quot;rows,&quot;,
      length(unique(df$Gene)), &quot;unique genes\n&quot;)
  
  # Global test across comparisons
  kw &lt;- kruskal.test(AbsFC ~ Comparisons, data = df)
  
  pw_kw &lt;-pairwise.wilcox.test(
    df$AbsFC,
    df$Comparisons,
    p.adjust.method = &quot;BH&quot;)
  
  print(pw_kw)

  p &lt;- ggplot(df, aes(x = Comparisons, y = AbsFC)) +
    geom_boxplot(aes(fill = Comparisons)) +
    theme_bw() +
    xlab(&quot; &quot;) +
    ylab(&quot;AbslogFC&quot;) +
    ggtitle(
      paste0(
        cluster_name,
        &quot; RNA AbsFC Comparisons\nKruskal–Wallis p = &quot;,
        signif(kw$p.value, 3)
      )
    ) +
    theme(
      plot.title = element_text(size = rel(1.3), hjust = 0.5),
      axis.title = element_text(size = 15),
      axis.text.x = element_text(size = 8, angle = 15)
    )

  return(p)


}</code></pre>
<pre class="r"><code>Filt_RMG0_RNA_fc_NoD4_NoRep &lt;- Filt_RMG0_RNA_fc_NoD4 %&gt;% 
  dplyr::select(-(contains(&quot;R&quot;)))

RNA_Metadata_NoD4_NoRep &lt;- RNA_Metadata %&gt;% 
  dplyr::filter(Timepoint != &quot;Day4&quot;) %&gt;% 
  dplyr::slice(-43:-46)
rownames(RNA_Metadata_NoD4_NoRep) &lt;- colnames(Filt_RMG0_RNA_fc_NoD4_NoRep)
# saveRDS(RNA_Metadata_NoD4_NoRep,&quot;data/DGE/Species/RNA_Metadata_NoD4_NoRep.RDS&quot;)

RNA_Metadata_NoD4_NoRep$dgelist &lt;- c(rep(c(&quot;Human_Day0&quot;,&quot;Human_Day2&quot;,&quot;Human_Day5&quot;,&quot;Human_Day15&quot;,&quot;Human_Day30&quot;),7),
                                     rep(c(&quot;Chimp_Day0&quot;,&quot;Chimp_Day2&quot;,&quot;Chimp_Day5&quot;,&quot;Chimp_Day15&quot;,&quot;Chimp_Day30&quot;),7))
#create DGEList object
dge &lt;- DGEList(counts = Filt_RMG0_RNA_fc_NoD4_NoRep)
dge$samples$group &lt;- factor(RNA_Metadata_NoD4_NoRep$dgelist)
dge &lt;- calcNormFactors(dge, method = &quot;TMM&quot;)
dge$samples</code></pre>
<pre><code>                 group lib.size norm.factors
H28126_D0   Human_Day0 25932966    1.0496515
H28126_D2   Human_Day2 16307187    1.0711273
H28126_D5   Human_Day5 12085267    1.0445905
H28126_D15 Human_Day15 14451707    0.9894752
H28126_D30 Human_Day30 13411160    0.8533304
H17_D0      Human_Day0 12100627    0.9998587
H17_D2      Human_Day2  7735466    1.1195784
H17_D5      Human_Day5 13218704    1.0490514
H17_D15    Human_Day15 23020627    1.0219939
H17_D30    Human_Day30 14484164    0.8028698
H78_D0      Human_Day0 17271130    1.1057177
H78_D2      Human_Day2 17996336    1.0618670
H78_D5      Human_Day5 11595332    1.0645316
H78_D15    Human_Day15 10335012    0.9912379
H78_D30    Human_Day30 12500044    0.7905653
H20682_D0   Human_Day0 12098542    1.0515114
H20682_D2   Human_Day2  8796211    1.0642338
H20682_D5   Human_Day5 14503271    1.0527575
H20682_D15 Human_Day15 15791894    0.9758863
H20682_D30 Human_Day30 13461509    0.8528200
H22422_D0   Human_Day0 20440576    0.9944732
H22422_D2   Human_Day2 14838910    1.0566006
H22422_D5   Human_Day5 10907743    1.0225115
H22422_D15 Human_Day15 15168583    0.9582386
H22422_D30 Human_Day30 15804750    0.8691100
H21792_D0   Human_Day0 10664025    1.0943173
H21792_D2   Human_Day2 13902708    1.1051554
H21792_D5   Human_Day5 27139690    1.0715092
H21792_D15 Human_Day15 13059206    0.9711326
H21792_D30 Human_Day30 11476245    0.8685525
H24280_D0   Human_Day0 13110354    0.9724776
H24280_D2   Human_Day2 13373790    1.0723900
H24280_D5   Human_Day5 10998176    0.9793593
H24280_D15 Human_Day15 11398361    1.0029455
H24280_D30 Human_Day30  9240969    0.8196602
C3649_D0    Chimp_Day0 12362301    1.0885058
C3649_D2    Chimp_Day2 15320271    1.0656161
C3649_D5    Chimp_Day5 15077601    1.1092666
C3649_D15  Chimp_Day15 12193785    1.1458739
C3649_D30  Chimp_Day30 15078429    0.8009392
C4955_D0    Chimp_Day0 10296889    1.1308602
C4955_D2    Chimp_Day2 17054948    1.0633736
C4955_D5    Chimp_Day5 10840478    1.0796001
C4955_D15  Chimp_Day15 18693180    1.0028659
C4955_D30  Chimp_Day30 13238998    0.7818647
C3651_D0    Chimp_Day0 15922479    1.0501976
C3651_D2    Chimp_Day2 12300292    1.1312951
C3651_D5    Chimp_Day5 20234297    1.0499510
C3651_D15  Chimp_Day15  8876621    0.9274931
C3651_D30  Chimp_Day30 13436497    0.7268620
C40210_D0   Chimp_Day0 14352669    1.0386225
C40210_D2   Chimp_Day2  9014514    1.1470563
C40210_D5   Chimp_Day5  9386102    1.0700081
C40210_D15 Chimp_Day15  8160685    0.9414793
C40210_D30 Chimp_Day30 15173940    0.8257046
C8861_D0    Chimp_Day0  7071000    1.1748857
C8861_D2    Chimp_Day2 12368759    1.1332137
C8861_D5    Chimp_Day5 13171748    1.1242551
C8861_D15  Chimp_Day15 14882942    1.1183146
C8861_D30  Chimp_Day30  9269576    0.7552249
C40280_D0   Chimp_Day0 17712680    1.0615566
C40280_D2   Chimp_Day2  9954155    1.1163790
C40280_D5   Chimp_Day5  8982698    1.0982065
C40280_D15 Chimp_Day15  9515327    1.0407279
C40280_D30 Chimp_Day30 13459189    0.8195138
C3647_D0    Chimp_Day0 11139654    1.1253799
C3647_D2    Chimp_Day2 12750800    1.1019634
C3647_D5    Chimp_Day5 20390838    1.0649082
C3647_D15  Chimp_Day15 16054523    0.9676347
C3647_D30  Chimp_Day30 13897133    0.7387200</code></pre>
<pre class="r"><code>#dim(dge)
# saveRDS(dge, &quot;data/DGE/Species/RNA_dge_matrix.RDS&quot;)
#dge$samples</code></pre>
<pre class="r"><code>design1 &lt;- model.matrix(~ 0 + RNA_Metadata_NoD4_NoRep$dgelist)
colnames(design1) &lt;- gsub(&quot;RNA_Metadata_NoD4_NoRep\\$dgelist&quot;, &quot;&quot;, colnames(design1))
View(design1)

corfit &lt;- duplicateCorrelation(object = dge$counts, design = design1, block = RNA_Metadata_NoD4_NoRep$Individual)
v &lt;- voom(dge, design1, block = RNA_Metadata_NoD4_NoRep$Individual, correlation = corfit$consensus.correlation, plot = TRUE)</code></pre>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_Data_Model-1.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>fit &lt;- lmFit(v, design1, block = RNA_Metadata_NoD4_NoRep$Individual, correlation = corfit$consensus.correlation)

contrast_matrix &lt;- makeContrasts(
  HC_D0 = Chimp_Day0 - Human_Day0,
  HC_D2 = Chimp_Day2 - Human_Day2,
  HC_D5 = Chimp_Day5 - Human_Day5,
  HC_D15 = Chimp_Day15 - Human_Day15,
  HC_D30 = Chimp_Day30 - Human_Day30,
  levels = design1)

fit2 &lt;- contrasts.fit(fit,contrast_matrix)
fit2 &lt;- eBayes(fit2)

plotSA(fit2, main = &quot;RNA Model&quot;)</code></pre>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_Data_Model-2.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>results_summary &lt;- decideTests(fit2,
                               adjust.method = &quot;BH&quot;,
                               p.value = 0.05)
summary(results_summary)</code></pre>
<pre><code>       HC_D0 HC_D2 HC_D5 HC_D15 HC_D30
Down    3371  1944  2070   2578   3127
NotSig  8269 10744 10684   9768   8848
Up      3198  2150  2084   2492   2863</code></pre>
<pre class="r"><code># saveRDS(fit2,&quot;data/DGE/Species/fit2.RDS&quot;)</code></pre>
<pre class="r"><code>#Create Human TopTable. using listapply, then you can pull specific comp using the $
TopTables_RNA_names &lt;- colnames(fit2$coefficients) 

# colnames(fit2$coefficients)

TopTable_RNA_All &lt;- do.call(
  rbind,
  lapply(TopTables_RNA_names,function(ct) {
    
    tt &lt;- topTable(
      fit2,
      coef = ct,
      number = Inf,
      sort.by = &quot;p&quot;
    )
    
    #add identifiers
    tt$Gene &lt;- rownames(tt)
    tt$Comparisons &lt;- ct
    
    tt
  })
)

# TopTable_RNA_All$Comparisons
# saveRDS(TopTable_RNA_All,&quot;data/DGE/Species/TopTable_RNA_all.RDS&quot;)
head(TopTable_RNA_All)</code></pre>
<pre><code>                    logFC    AveExpr        t      P.Value    adj.P.Val
ENSG00000174231  6.604341 5.99612713 52.85392 9.143730e-55 1.356747e-50
ENSG00000184983 11.237428 0.75630938 44.99139 2.230381e-50 1.654720e-46
ENSG00000272540  9.859124 0.05875218 39.97301 3.493933e-47 1.728099e-43
ENSG00000177370  9.890876 0.11451324 39.47648 7.575533e-47 2.252354e-43
ENSG00000231074 10.811687 0.89083894 39.47528 7.589817e-47 2.252354e-43
ENSG00000108953  5.382156 7.03619706 38.02789 7.632500e-46 1.887517e-42
                        B            Gene Comparisons
ENSG00000174231 112.74454 ENSG00000174231       HC_D0
ENSG00000184983  99.69885 ENSG00000184983       HC_D0
ENSG00000272540  93.52547 ENSG00000272540       HC_D0
ENSG00000177370  92.84590 ENSG00000177370       HC_D0
ENSG00000231074  92.94400 ENSG00000231074       HC_D0
ENSG00000108953  94.02074 ENSG00000108953       HC_D0</code></pre>
<pre class="r"><code>Comparisons_names &lt;- unique(TopTable_RNA_All$Comparisons)
# saveRDS(Comparisons_names,&quot;data/DGE/Species/Comparisons_names_Trajectory.RDS&quot;)
plot &lt;- list()
for (n in Comparisons_names) {

plot[[n]] &lt;- TopTable_RNA_All %&gt;% 
  dplyr::filter(Comparisons == paste0(n)) %&gt;% 
  generate_volcano_plot(.,n)
}

for (n in Comparisons_names) {
  print(plot[[n]])
        }</code></pre>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_VolcancoPlots-1.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNASeq_DGE_Species_TopTable_VolcancoPlots-1">
Past versions of RNASeq_DGE_Species_TopTable_VolcancoPlots-1.png
</button>
</p>
<div id="fig-RNASeq_DGE_Species_TopTable_VolcancoPlots-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/b0ab413db9b92315393162a697a43c146d357b7d/docs/figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_VolcancoPlots-1.png" target="_blank">b0ab413</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_VolcancoPlots-2.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNASeq_DGE_Species_TopTable_VolcancoPlots-2">
Past versions of RNASeq_DGE_Species_TopTable_VolcancoPlots-2.png
</button>
</p>
<div id="fig-RNASeq_DGE_Species_TopTable_VolcancoPlots-2"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/b0ab413db9b92315393162a697a43c146d357b7d/docs/figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_VolcancoPlots-2.png" target="_blank">b0ab413</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_VolcancoPlots-3.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNASeq_DGE_Species_TopTable_VolcancoPlots-3">
Past versions of RNASeq_DGE_Species_TopTable_VolcancoPlots-3.png
</button>
</p>
<div id="fig-RNASeq_DGE_Species_TopTable_VolcancoPlots-3"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/b0ab413db9b92315393162a697a43c146d357b7d/docs/figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_VolcancoPlots-3.png" target="_blank">b0ab413</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_VolcancoPlots-4.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNASeq_DGE_Species_TopTable_VolcancoPlots-4">
Past versions of RNASeq_DGE_Species_TopTable_VolcancoPlots-4.png
</button>
</p>
<div id="fig-RNASeq_DGE_Species_TopTable_VolcancoPlots-4"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/b0ab413db9b92315393162a697a43c146d357b7d/docs/figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_VolcancoPlots-4.png" target="_blank">b0ab413</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_VolcancoPlots-5.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNASeq_DGE_Species_TopTable_VolcancoPlots-5">
Past versions of RNASeq_DGE_Species_TopTable_VolcancoPlots-5.png
</button>
</p>
<div id="fig-RNASeq_DGE_Species_TopTable_VolcancoPlots-5"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/b0ab413db9b92315393162a697a43c146d357b7d/docs/figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_VolcancoPlots-5.png" target="_blank">b0ab413</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>days &lt;- c(&quot;D0&quot;, &quot;D2&quot;, &quot;D5&quot;, &quot;D15&quot;, &quot;D30&quot;)

RNA_sets &lt;- map(
  days,
  ~ {
    df &lt;- TopTable_RNA_All %&gt;%
      filter(Comparisons == paste0(&quot;HC_&quot;, .x))

    list(
      DEGs = df %&gt;% filter(adj.P.Val &lt; 0.05),
      NonDEGs = df %&gt;% filter(adj.P.Val &gt;= 0.05)
    )
  }
)

names(RNA_sets) &lt;- paste0(&quot;Day&quot;, sub(&quot;D&quot;, &quot;&quot;, days), &quot;_HC&quot;)

# saveRDS(RNA_sets,&quot;data/DGE/Species/RNA_DEG_Non_DEG_sets_05.RDS&quot;)</code></pre>
<pre class="r"><code>all_genes &lt;- unique(TopTable_RNA_All$Gene)
  
DEG_Lists &lt;- list(
  Day0_DEGs_HC = RNA_sets$Day0_HC$DEGs,       # TopTable data.frame
  Day2_DEGs_HC = RNA_sets$Day2_HC$DEGs,
  Day5_DEGs_HC = RNA_sets$Day5_HC$DEGs,
  Day15_DEGs_HC = RNA_sets$Day15_HC$DEGs,
  Day30_DEGs_HC = RNA_sets$Day30_HC$DEGs
)

fit_list &lt;- list()

for (name in names(DEG_Lists)) {

  # DEG gene vector
  deg_vector &lt;- unique(DEG_Lists[[name]]$Gene)

  # Euler with containment
  fit_list[[name]] &lt;- euler(list(
    AllGenes = all_genes,
    DEGs = deg_vector
  ))

  # Plot
  print(
    plot(
      fit_list[[name]],
      fills = list(
        fill = c(&quot;grey85&quot;, &quot;firebrick&quot;),
        alpha = 0.6
      ),
      labels = c(&quot;NonDEGs&quot;, &quot;DEGs&quot;),
      quantities = TRUE,
      main = paste0(name, &quot; (p &lt; 0.05)&quot;)
    )
  )
}</code></pre>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Subsetting_DEGs0.05_Eulers-1.png" alt="" width="672" style="display: block; margin: auto;" /><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Subsetting_DEGs0.05_Eulers-2.png" alt="" width="672" style="display: block; margin: auto;" /><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Subsetting_DEGs0.05_Eulers-3.png" alt="" width="672" style="display: block; margin: auto;" /><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Subsetting_DEGs0.05_Eulers-4.png" alt="" width="672" style="display: block; margin: auto;" /><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Subsetting_DEGs0.05_Eulers-5.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>####proportion bar plots####
#first factor the timepoints so that they are in the right order
df &lt;- data.frame(
  Day = c(&quot;Day0&quot;, &quot;Day2&quot;, &quot;Day5&quot;, &quot;Day15&quot;, &quot;Day30&quot;),
  DEGs = c(6569, 4094, 4145, 5070, 5990),
  Non_DEGs = c(8269, 10744, 10684, 9768, 8848)
)

# Convert to long format for ggplot
df_long &lt;- df %&gt;%
  pivot_longer(cols = c(DEGs, Non_DEGs),
               names_to = &quot;Type&quot;,
               values_to = &quot;Count&quot;) %&gt;%
  group_by(Day) %&gt;%
  mutate(Proportion = Count / sum(Count))  # calculate proportions

df_long$Day &lt;- factor(df_long$Day,levels = c(&quot;Day0&quot;, &quot;Day2&quot;, &quot;Day5&quot;, &quot;Day15&quot;, &quot;Day30&quot;),ordered = TRUE)

# df_long

ggplot(df_long, aes(x = Day, y = Proportion, fill = Type)) +
  geom_bar(stat = &quot;identity&quot;) +
  scale_y_continuous(labels = scales::percent_format()) +  # show % on y-axis
  scale_fill_manual(values = c(&quot;DEGs&quot; = &quot;#E41A1C&quot;, &quot;Non_DEGs&quot; = &quot;#377EB8&quot;)) + # custom colors
  labs(y = &quot;Proportion&quot;, x = &quot;&quot;, fill = &quot;&quot;,title = &quot;Proportion of Species DEGs p &lt; 0.05&quot;) +
  theme_minimal(base_size = 14)</code></pre>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Subsetting0.5_barplot-1.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>days &lt;- c(&quot;D0&quot;, &quot;D2&quot;, &quot;D5&quot;, &quot;D15&quot;, &quot;D30&quot;)

RNA_sets_0.1 &lt;- map(
  days,
  ~ {
    df &lt;- TopTable_RNA_All %&gt;%
      filter(Comparisons == paste0(&quot;HC_&quot;, .x))

    list(
      DEGs = df %&gt;% filter(adj.P.Val &lt; 0.1),
      NonDEGs = df %&gt;% filter(adj.P.Val &gt;= 0.1)
    )
  }
)

names(RNA_sets_0.1) &lt;- paste0(&quot;Day&quot;, sub(&quot;D&quot;, &quot;&quot;, days), &quot;_HC&quot;)

# saveRDS(RNA_sets_0.1,&quot;data/DGE/Species/RNA_DEG_Non_DEG_sets_05.RDS&quot;)</code></pre>
<pre class="r"><code>all_genes &lt;- unique(TopTable_RNA_All$Gene)
  
DEG_Lists &lt;- list(
  Day0_DEGs_HC = RNA_sets_0.1$Day0_HC$DEGs,       # TopTable data.frame
  Day2_DEGs_HC = RNA_sets_0.1$Day2_HC$DEGs,
  Day5_DEGs_HC = RNA_sets_0.1$Day5_HC$DEGs,
  Day15_DEGs_HC = RNA_sets_0.1$Day15_HC$DEGs,
  Day30_DEGs_HC = RNA_sets_0.1$Day30_HC$DEGs
)

fit_list &lt;- list()

for (name in names(DEG_Lists)) {

  # DEG gene vector
  deg_vector &lt;- unique(DEG_Lists[[name]]$Gene)

  # Euler with containment
  fit_list[[name]] &lt;- euler(list(
    AllGenes = all_genes,
    DEGs = deg_vector
  ))

  # Plot
  print(
    plot(
      fit_list[[name]],
      fills = list(
        fill = c(&quot;grey85&quot;, &quot;firebrick&quot;),
        alpha = 0.6
      ),
      labels = c(&quot;NonDEGs&quot;, &quot;DEGs&quot;),
      quantities = TRUE,
      main = paste0(name, &quot; (p &lt; 0.1)&quot;)
    )
  )
}</code></pre>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Subsetting_DEGs0.1_Eulers-1.png" alt="" width="672" style="display: block; margin: auto;" /><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Subsetting_DEGs0.1_Eulers-2.png" alt="" width="672" style="display: block; margin: auto;" /><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Subsetting_DEGs0.1_Eulers-3.png" alt="" width="672" style="display: block; margin: auto;" /><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Subsetting_DEGs0.1_Eulers-4.png" alt="" width="672" style="display: block; margin: auto;" /><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Subsetting_DEGs0.1_Eulers-5.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>####proportion bar plots####
#first factor the timepoints so that they are in the right order
df &lt;- data.frame(
  Day = c(&quot;Day0&quot;, &quot;Day2&quot;, &quot;Day5&quot;, &quot;Day15&quot;, &quot;Day30&quot;),
  DEGs = c(7539, 5012, 5216, 6103, 7040),
  Non_DEGs = c(7299, 9826, 9622, 8735, 7798)
)

# Convert to long format for ggplot
df_long &lt;- df %&gt;%
  pivot_longer(cols = c(DEGs, Non_DEGs),
               names_to = &quot;Type&quot;,
               values_to = &quot;Count&quot;) %&gt;%
  group_by(Day) %&gt;%
  mutate(Proportion = Count / sum(Count))  # calculate proportions

df_long$Day &lt;- factor(df_long$Day,levels = c(&quot;Day0&quot;, &quot;Day2&quot;, &quot;Day5&quot;, &quot;Day15&quot;, &quot;Day30&quot;),ordered = TRUE)

# df_long

ggplot(df_long, aes(x = Day, y = Proportion, fill = Type)) +
  geom_bar(stat = &quot;identity&quot;) +
  scale_y_continuous(labels = scales::percent_format()) +  # show % on y-axis
  scale_fill_manual(values = c(&quot;DEGs&quot; = &quot;#E41A1C&quot;, &quot;Non_DEGs&quot; = &quot;#377EB8&quot;)) + # custom colors
  labs(y = &quot;Proportion&quot;, x = &quot;&quot;, fill = &quot;&quot;, title = &quot;Proportion of Species DEGs p &lt; 0.1&quot;) +
  theme_minimal(base_size = 14)</code></pre>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Subsetting0.1_barplot-1.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>TopTable_RNA_All &lt;- readRDS(&quot;data/DGE/Species/TopTable_RNA_all.RDS&quot;)
all_genes &lt;- readRDS(&quot;data/DGE/Species/ExpressedGenes.RDS&quot;)
deg_genes &lt;- readRDS(&quot;data/DGE/Species/DEG_AtLeastOne.RDS&quot;)

Gene_list &lt;- list(
  All_genes  = all_genes,
  DEGs = deg_genes
)

cluster_plots &lt;- lapply(names(Gene_list), function(clust_name) {
  plot_cluster_fc_sig(Gene_list[[clust_name]], clust_name, TopTable_RNA_All)
})</code></pre>
<pre><code>All_genes : 74190 rows, 14838 unique genes

    Pairwise comparisons using Wilcoxon rank sum test with continuity correction 

data:  df$logFC and df$Comparisons 

       HC_D0   HC_D2   HC_D5   HC_D15 
HC_D2  &lt; 2e-16 -       -       -      
HC_D5  2.8e-10 0.00025 -       -      
HC_D15 3.7e-13 2.1e-05 0.61867 -      
HC_D30 2.7e-08 1.0e-08 0.06482 0.03293

P value adjustment method: BH 
DEGs : 58240 rows, 11648 unique genes

    Pairwise comparisons using Wilcoxon rank sum test with continuity correction 

data:  df$logFC and df$Comparisons 

       HC_D0   HC_D2   HC_D5   HC_D15 
HC_D2  &lt; 2e-16 -       -       -      
HC_D5  1.2e-09 0.00561 -       -      
HC_D15 &lt; 2e-16 0.06704 0.26136 -      
HC_D30 2.6e-11 0.00063 0.73944 0.03392

P value adjustment method: BH </code></pre>
<pre class="r"><code>names(cluster_plots) &lt;- names(Gene_list)

for (p in cluster_plots) print(p)</code></pre>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Subsetting_FC-1.png" alt="" width="672" style="display: block; margin: auto;" /><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Subsetting_FC-2.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Day0_AllGenes &lt;- TopTable_RNA_All %&gt;% 
  dplyr::filter(Comparisons == &quot;HC_D0&quot;)
print(&quot;Day 0 Average logFC&quot;)</code></pre>
<pre><code>[1] &quot;Day 0 Average logFC&quot;</code></pre>
<pre class="r"><code>print(mean(Day0_AllGenes$logFC))</code></pre>
<pre><code>[1] -0.04176301</code></pre>
<pre class="r"><code>Day2_AllGenes &lt;- TopTable_RNA_All %&gt;% 
  dplyr::filter(Comparisons == &quot;HC_D2&quot;)
print(&quot;Day 2 Average logFC&quot;)</code></pre>
<pre><code>[1] &quot;Day 2 Average logFC&quot;</code></pre>
<pre class="r"><code>print(mean(Day2_AllGenes$logFC))</code></pre>
<pre><code>[1] 0.1291112</code></pre>
<pre class="r"><code>Day5_AllGenes &lt;- TopTable_RNA_All %&gt;% 
  dplyr::filter(Comparisons == &quot;HC_D5&quot;)
print(&quot;Day 5 Average logFC&quot;)</code></pre>
<pre><code>[1] &quot;Day 5 Average logFC&quot;</code></pre>
<pre class="r"><code>print(mean(Day5_AllGenes$logFC))</code></pre>
<pre><code>[1] 0.06418101</code></pre>
<pre class="r"><code>Day15_AllGenes &lt;- TopTable_RNA_All %&gt;% 
  dplyr::filter(Comparisons == &quot;HC_D15&quot;)
print(&quot;Day 15 Average logFC&quot;)</code></pre>
<pre><code>[1] &quot;Day 15 Average logFC&quot;</code></pre>
<pre class="r"><code>print(mean(Day15_AllGenes$logFC))</code></pre>
<pre><code>[1] 0.1353744</code></pre>
<pre class="r"><code>Day30_AllGenes &lt;- TopTable_RNA_All %&gt;% 
  dplyr::filter(Comparisons == &quot;HC_D30&quot;)
print(&quot;Day 30 Average logFC&quot;)</code></pre>
<pre><code>[1] &quot;Day 30 Average logFC&quot;</code></pre>
<pre class="r"><code>print(mean(Day30_AllGenes$logFC))</code></pre>
<pre><code>[1] 0.10031</code></pre>
<pre class="r"><code>all_genes &lt;- readRDS(&quot;data/DGE/Species/ExpressedGenes.RDS&quot;)
deg_genes &lt;- readRDS(&quot;data/DGE/Species/DEG_AtLeastOne.RDS&quot;)

TopTable_RNA_All$AbsFC &lt;- abs(TopTable_RNA_All$logFC)

Gene_list &lt;- list(
  All_genes  = all_genes,
  DEGs = deg_genes
)

cluster_plots &lt;- lapply(names(Gene_list), function(clust_name) {
  plot_cluster_absfc_sig(Gene_list[[clust_name]], clust_name, TopTable_RNA_All)
})</code></pre>
<pre><code>All_genes : 74190 rows, 14838 unique genes

    Pairwise comparisons using Wilcoxon rank sum test with continuity correction 

data:  df$AbsFC and df$Comparisons 

       HC_D0   HC_D2   HC_D5   HC_D15 
HC_D2  &lt; 2e-16 -       -       -      
HC_D5  &lt; 2e-16 0.46    -       -      
HC_D15 &lt; 2e-16 3.9e-15 &lt; 2e-16 -      
HC_D30 0.18    &lt; 2e-16 &lt; 2e-16 &lt; 2e-16

P value adjustment method: BH 
DEGs : 58240 rows, 11648 unique genes

    Pairwise comparisons using Wilcoxon rank sum test with continuity correction 

data:  df$AbsFC and df$Comparisons 

       HC_D0  HC_D2  HC_D5  HC_D15
HC_D2  &lt;2e-16 -      -      -     
HC_D5  &lt;2e-16 0.23   -      -     
HC_D15 &lt;2e-16 &lt;2e-16 &lt;2e-16 -     
HC_D30 0.22   &lt;2e-16 &lt;2e-16 &lt;2e-16

P value adjustment method: BH </code></pre>
<pre class="r"><code>names(cluster_plots) &lt;- names(Gene_list)

for (p in cluster_plots) print(p)</code></pre>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Subsetting_AbsFC-1.png" alt="" width="672" style="display: block; margin: auto;" /><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Subsetting_AbsFC-2.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Day0_AllGenes &lt;- TopTable_RNA_All %&gt;% 
  dplyr::filter(Comparisons == &quot;HC_D0&quot;)
print(&quot;Day 0 Average AbslogFC&quot;)</code></pre>
<pre><code>[1] &quot;Day 0 Average AbslogFC&quot;</code></pre>
<pre class="r"><code>print(mean(Day0_AllGenes$AbsFC))</code></pre>
<pre><code>[1] 0.7933367</code></pre>
<pre class="r"><code>Day2_AllGenes &lt;- TopTable_RNA_All %&gt;% 
  dplyr::filter(Comparisons == &quot;HC_D2&quot;)
print(&quot;Day 2 Average AbslogFC&quot;)</code></pre>
<pre><code>[1] &quot;Day 2 Average AbslogFC&quot;</code></pre>
<pre class="r"><code>print(mean(Day2_AllGenes$AbsFC))</code></pre>
<pre><code>[1] 0.6770691</code></pre>
<pre class="r"><code>Day5_AllGenes &lt;- TopTable_RNA_All %&gt;% 
  dplyr::filter(Comparisons == &quot;HC_D5&quot;)
print(&quot;Day 5 Average AbslogFC&quot;)</code></pre>
<pre><code>[1] &quot;Day 5 Average AbslogFC&quot;</code></pre>
<pre class="r"><code>print(mean(Day5_AllGenes$AbsFC))</code></pre>
<pre><code>[1] 0.665505</code></pre>
<pre class="r"><code>Day15_AllGenes &lt;- TopTable_RNA_All %&gt;% 
  dplyr::filter(Comparisons == &quot;HC_D15&quot;)
print(&quot;Day 15 Average AbslogFC&quot;)</code></pre>
<pre><code>[1] &quot;Day 15 Average AbslogFC&quot;</code></pre>
<pre class="r"><code>print(mean(Day15_AllGenes$AbsFC))</code></pre>
<pre><code>[1] 0.7349379</code></pre>
<pre class="r"><code>Day30_AllGenes &lt;- TopTable_RNA_All %&gt;% 
  dplyr::filter(Comparisons == &quot;HC_D30&quot;)
print(&quot;Day 30 Average AbslogFC&quot;)</code></pre>
<pre><code>[1] &quot;Day 30 Average AbslogFC&quot;</code></pre>
<pre class="r"><code>print(mean(Day30_AllGenes$AbsFC))</code></pre>
<pre><code>[1] 0.8157532</code></pre>
<pre class="r"><code># Your DEGs per day (already defined)

sets &lt;- list(
  Day0_DEGs_HC = RNA_sets$Day0_HC$DEGs$Gene,       # TopTable data.frame
  Day2_DEGs_HC = RNA_sets$Day2_HC$DEGs$Gene,
  Day5_DEGs_HC = RNA_sets$Day5_HC$DEGs$Gene,
  Day15_DEGs_HC = RNA_sets$Day15_HC$DEGs$Gene,
  Day30_DEGs_HC = RNA_sets$Day30_HC$DEGs$Gene
)</code></pre>
<pre class="r"><code># Create a 5-way Venn diagram
p &lt;- ggVennDiagram(sets,
                   label_alpha = 0,       # makes background of labels transparent
                   label = &quot;count&quot;) +     # show counts in intersections
  scale_fill_gradient(low = &quot;skyblue&quot;, high = &quot;navy&quot;) + # optional color gradient
  theme(legend.position = &quot;none&quot;) +                     # hide legend
  ggtitle(&quot;5-way DEGs&quot;)

# Plot
print(p)</code></pre>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Venn_ggplot_plot-1.png" alt="" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNASeq_DGE_Species_TopTable_Venn_ggplot_plot-1">
Past versions of RNASeq_DGE_Species_TopTable_Venn_ggplot_plot-1.png
</button>
</p>
<div id="fig-RNASeq_DGE_Species_TopTable_Venn_ggplot_plot-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/b0ab413db9b92315393162a697a43c146d357b7d/docs/figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Species_TopTable_Venn_ggplot_plot-1.png" target="_blank">b0ab413</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Helper function to get intersections
get_intersections &lt;- function(sets_list) {
  set_names &lt;- names(sets_list)
  n &lt;- length(sets_list)
  result &lt;- list()
  
  # Loop over all combination lengths (1 to n)
  for (k in 1:n) {
    combos &lt;- combn(set_names, k, simplify = FALSE)
    for (combo in combos) {
      # Start with first set in combo
      genes &lt;- sets_list[[combo[1]]]
      # Intersect with the rest
      if (length(combo) &gt; 1) {
        for (s in combo[-1]) {
          genes &lt;- intersect(genes, sets_list[[s]])
        }
      }
      # Remove genes that appear in other sets outside the combo (exclusive)
      other_sets &lt;- setdiff(set_names, combo)
      if (length(other_sets) &gt; 0) {
        for (s in other_sets) {
          genes &lt;- setdiff(genes, sets_list[[s]])
        }
      }
      # Save if any genes remain
      if (length(genes) &gt; 0) {
        result[[paste(combo, collapse = &quot;&amp;&quot;)]] &lt;- genes
      }
    }
  }
  return(result)
}

# Compute all intersections
all_intersections &lt;- get_intersections(sets)
# saveRDS(all_intersections,&quot;data/DGE/Species/all_intersections.RDS&quot;)

# # Example: genes only in Day0
# all_intersections$Day0

# # Example: genes only in Day0 &amp; Day2
# all_intersections$`Day0&amp;Day2`</code></pre>
<pre class="r"><code>H_comp &lt;- &quot;HC_D0&quot;
C_comp &lt;- &quot;HC_D2&quot;

df_wide &lt;- TopTable_RNA_All %&gt;%
  filter(Comparisons %in% c(H_comp, C_comp)) %&gt;%
  dplyr::select(Gene, Comparisons, logFC) %&gt;%
  group_by(Gene, Comparisons) %&gt;%
  summarize(logFC = mean(logFC), .groups = &quot;drop&quot;) %&gt;%
  pivot_wider(names_from = Comparisons, values_from = logFC) %&gt;%
  filter(!is.na(.data[[H_comp]]), !is.na(.data[[C_comp]]))

# head(df_wide)
# nrow(df_wide)

fit &lt;- lm(df_wide[[C_comp]] ~ df_wide[[H_comp]])
summary(fit)</code></pre>
<pre><code>
Call:
lm(formula = df_wide[[C_comp]] ~ df_wide[[H_comp]])

Residuals:
    Min      1Q  Median      3Q     Max 
-6.2637 -0.4703 -0.1342  0.3374  6.5623 

Coefficients:
                  Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)       0.160214   0.006291   25.47   &lt;2e-16 ***
df_wide[[H_comp]] 0.744745   0.004702  158.41   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.7659 on 14836 degrees of freedom
Multiple R-squared:  0.6284,    Adjusted R-squared:  0.6284 
F-statistic: 2.509e+04 on 1 and 14836 DF,  p-value: &lt; 2.2e-16</code></pre>
<pre class="r"><code>r2 &lt;- summary(fit)$r.squared
pval &lt;- summary(fit)$coefficients[2, &quot;Pr(&gt;|t|)&quot;]

# r2
# pval

# format(pval, scientific = TRUE, digits =3)

x_pos &lt;- max(df_wide[[H_comp]]) * 0.95      # right side
x_neg &lt;- min(df_wide[[H_comp]]) * 0.5      # left side

y_pos &lt;- max(df_wide[[C_comp]]) * 0.95

ggplot(df_wide, aes(x = .data[[H_comp]], y = .data[[C_comp]])) +
  geom_point(size = 3) +
  geom_smooth(method = &quot;lm&quot;, se = FALSE, linetype = &quot;dashed&quot;) +
  geom_hline(yintercept = 0, linetype = &quot;dashed&quot;) +
  geom_vline(xintercept = 0, linetype = &quot;dashed&quot;) +
  annotate(
    &quot;text&quot;,
    x = x_neg,
    y = y_pos,
    label = paste0(
      &quot;R² = &quot;, round(r2, 3),
      &quot;\np = &quot;, format(pval, scientific = TRUE, digits = 3)
    ),
    hjust = 1,
    vjust = 1
  ) +
  theme_bw() +
  labs(
    x = H_comp,
    y = C_comp,
    title = paste0(H_comp, &quot; vs &quot;, C_comp)
  )</code></pre>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Clusters_FC_Quadrants_0_2-1.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>H_comp &lt;- &quot;HC_D2&quot;
C_comp &lt;- &quot;HC_D5&quot;

df_wide &lt;- TopTable_RNA_All %&gt;%
  filter(Comparisons %in% c(H_comp, C_comp)) %&gt;%
  dplyr::select(Gene, Comparisons, logFC) %&gt;%
  group_by(Gene, Comparisons) %&gt;%
  summarize(logFC = mean(logFC), .groups = &quot;drop&quot;) %&gt;%
  pivot_wider(names_from = Comparisons, values_from = logFC) %&gt;%
  filter(!is.na(.data[[H_comp]]), !is.na(.data[[C_comp]]))

# head(df_wide)
# nrow(df_wide)

fit &lt;- lm(df_wide[[C_comp]] ~ df_wide[[H_comp]])
summary(fit)</code></pre>
<pre><code>
Call:
lm(formula = df_wide[[C_comp]] ~ df_wide[[H_comp]])

Residuals:
    Min      1Q  Median      3Q     Max 
-6.0578 -0.2940  0.0238  0.3482  8.1083 

Coefficients:
                   Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)       -0.040325   0.005812  -6.938 4.14e-12 ***
df_wide[[H_comp]]  0.809428   0.004602 175.891  &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.7043 on 14836 degrees of freedom
Multiple R-squared:  0.6759,    Adjusted R-squared:  0.6759 
F-statistic: 3.094e+04 on 1 and 14836 DF,  p-value: &lt; 2.2e-16</code></pre>
<pre class="r"><code>r2 &lt;- summary(fit)$r.squared
pval &lt;- summary(fit)$coefficients[2, &quot;Pr(&gt;|t|)&quot;]

# r2
# pval

# format(pval, scientific = TRUE, digits = 3)

library(ggplot2)

x_pos &lt;- max(df_wide[[H_comp]]) * 0.95      # right side
x_neg &lt;- min(df_wide[[H_comp]]) * 0.5      # left side

y_pos &lt;- max(df_wide[[C_comp]]) * 0.95

ggplot(df_wide, aes(x = .data[[H_comp]], y = .data[[C_comp]])) +
  geom_point(size = 3) +
  geom_smooth(method = &quot;lm&quot;, se = FALSE, linetype = &quot;dashed&quot;) +
  geom_hline(yintercept = 0, linetype = &quot;dashed&quot;) +
  geom_vline(xintercept = 0, linetype = &quot;dashed&quot;) +
  annotate(
    &quot;text&quot;,
    x = x_neg,
    y = y_pos,
    label = paste0(
      &quot;R² = &quot;, round(r2, 3),
      &quot;\np = &quot;, format(pval, scientific = TRUE, digits = 3)
    ),
    hjust = 1,
    vjust = 1
  ) +
  theme_bw() +
  labs(
    x = H_comp,
    y = C_comp,
    title = paste0(H_comp, &quot; vs &quot;, C_comp)
  )</code></pre>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Clusters_FC_Quadrants_2_5-1.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>H_comp &lt;- &quot;HC_D5&quot;
C_comp &lt;- &quot;HC_D15&quot;

df_wide &lt;- TopTable_RNA_All %&gt;%
  filter(Comparisons %in% c(H_comp, C_comp)) %&gt;%
  dplyr::select(Gene, Comparisons, logFC) %&gt;%
  group_by(Gene, Comparisons) %&gt;%
  summarize(logFC = mean(logFC), .groups = &quot;drop&quot;) %&gt;%
  pivot_wider(names_from = Comparisons, values_from = logFC) %&gt;%
  filter(!is.na(.data[[H_comp]]), !is.na(.data[[C_comp]]))

# head(df_wide)
# nrow(df_wide)

fit &lt;- lm(df_wide[[C_comp]] ~ df_wide[[H_comp]])
summary(fit)</code></pre>
<pre><code>
Call:
lm(formula = df_wide[[C_comp]] ~ df_wide[[H_comp]])

Residuals:
    Min      1Q  Median      3Q     Max 
-5.7407 -0.4346 -0.0991  0.2863  7.6044 

Coefficients:
                  Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)       0.079014   0.006138   12.87   &lt;2e-16 ***
df_wide[[H_comp]] 0.878150   0.004956  177.20   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.7467 on 14836 degrees of freedom
Multiple R-squared:  0.6791,    Adjusted R-squared:  0.6791 
F-statistic: 3.14e+04 on 1 and 14836 DF,  p-value: &lt; 2.2e-16</code></pre>
<pre class="r"><code>r2 &lt;- summary(fit)$r.squared
pval &lt;- summary(fit)$coefficients[2, &quot;Pr(&gt;|t|)&quot;]

# r2
# pval

# format(pval, scientific = TRUE, digits = 3)

x_pos &lt;- max(df_wide[[H_comp]]) * 0.95      # right side
x_neg &lt;- min(df_wide[[H_comp]]) * 0.5      # left side

y_pos &lt;- max(df_wide[[C_comp]]) * 0.95

ggplot(df_wide, aes(x = .data[[H_comp]], y = .data[[C_comp]])) +
  geom_point(size = 3) +
  geom_smooth(method = &quot;lm&quot;, se = FALSE, linetype = &quot;dashed&quot;) +
  geom_hline(yintercept = 0, linetype = &quot;dashed&quot;) +
  geom_vline(xintercept = 0, linetype = &quot;dashed&quot;) +
  annotate(
    &quot;text&quot;,
    x = x_neg,
    y = y_pos,
    label = paste0(
      &quot;R² = &quot;, round(r2, 3),
      &quot;\np = &quot;, format(pval, scientific = TRUE, digits = 3)
    ),
    hjust = 1,
    vjust = 1
  ) +
  theme_bw() +
  labs(
    x = H_comp,
    y = C_comp,
    title = paste0(H_comp, &quot; vs &quot;, C_comp)
  )</code></pre>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Clusters_FC_Quadrants_5_15-1.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>H_comp &lt;- &quot;HC_D15&quot;
C_comp &lt;- &quot;HC_D30&quot;

df_wide &lt;- TopTable_RNA_All %&gt;%
  filter(Comparisons %in% c(H_comp, C_comp)) %&gt;%
  dplyr::select(Gene, Comparisons, logFC) %&gt;%
  group_by(Gene, Comparisons) %&gt;%
  summarize(logFC = mean(logFC), .groups = &quot;drop&quot;) %&gt;%
  pivot_wider(names_from = Comparisons, values_from = logFC) %&gt;%
  filter(!is.na(.data[[H_comp]]), !is.na(.data[[C_comp]]))

# head(df_wide)
# nrow(df_wide)

fit &lt;- lm(df_wide[[C_comp]] ~ df_wide[[H_comp]])
summary(fit)</code></pre>
<pre><code>
Call:
lm(formula = df_wide[[C_comp]] ~ df_wide[[H_comp]])

Residuals:
    Min      1Q  Median      3Q     Max 
-8.5173 -0.3344 -0.0048  0.3363  6.0096 

Coefficients:
                   Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)       -0.014621   0.006633  -2.204   0.0275 *  
df_wide[[H_comp]]  0.848989   0.005006 169.611   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.8037 on 14836 degrees of freedom
Multiple R-squared:  0.6598,    Adjusted R-squared:  0.6597 
F-statistic: 2.877e+04 on 1 and 14836 DF,  p-value: &lt; 2.2e-16</code></pre>
<pre class="r"><code>r2 &lt;- summary(fit)$r.squared
pval &lt;- summary(fit)$coefficients[2, &quot;Pr(&gt;|t|)&quot;]

# r2
# pval

# format(pval, scientific = TRUE, digits = 3)

x_pos &lt;- max(df_wide[[H_comp]]) * 0.95      # right side
x_neg &lt;- min(df_wide[[H_comp]]) * 0.5      # left side

y_pos &lt;- max(df_wide[[C_comp]]) * 0.95

ggplot(df_wide, aes(x = .data[[H_comp]], y = .data[[C_comp]])) +
  geom_point(size = 3) +
  geom_smooth(method = &quot;lm&quot;, se = FALSE, linetype = &quot;dashed&quot;) +
  geom_hline(yintercept = 0, linetype = &quot;dashed&quot;) +
  geom_vline(xintercept = 0, linetype = &quot;dashed&quot;) +
  annotate(
    &quot;text&quot;,
    x = x_neg,
    y = y_pos,
    label = paste0(
      &quot;R² = &quot;, round(r2, 3),
      &quot;\np = &quot;, format(pval, scientific = TRUE, digits = 3)
    ),
    hjust = 1,
    vjust = 1
  ) +
  theme_bw() +
  labs(
    x = H_comp,
    y = C_comp,
    title = paste0(H_comp, &quot; vs &quot;, C_comp)
  )</code></pre>
<p><img src="figure/RNA_DGE_Comp_Species_Ensemble.Rmd/RNASeq_DGE_Clusters_FC_Quadrants_15_30-1.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># git -&gt; commit all changes
# git -&gt; push
# wflow_publish(&quot;analysis/RNA_DGE_Comp_Species_Ensemble.Rmd&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.5.1 (2025-06-13 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 26100)

Matrix products: default
  LAPACK version 3.12.1

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggsignif_0.6.4       edgeR_4.6.3          limma_3.64.3        
 [4] ggrastr_1.0.2        ggVennDiagram_1.5.7  org.Hs.eg.db_3.21.0 
 [7] AnnotationDbi_1.70.0 IRanges_2.42.0       S4Vectors_0.46.0    
[10] Biobase_2.68.0       BiocGenerics_0.54.1  generics_0.1.4      
[13] eulerr_7.0.4         patchwork_1.3.2      ggfortify_0.4.19    
[16] ggrepel_0.9.6        readxl_1.4.5         lubridate_1.9.5     
[19] forcats_1.0.1        stringr_1.6.0        dplyr_1.1.4         
[22] purrr_1.2.1          readr_2.1.6          tidyr_1.3.2         
[25] tibble_3.3.1         ggplot2_4.0.2        tidyverse_2.0.0     
[28] workflowr_1.7.2     

loaded via a namespace (and not attached):
 [1] DBI_1.2.3               gridExtra_2.3           rlang_1.1.7            
 [4] magrittr_2.0.4          git2r_0.36.2            otel_0.2.0             
 [7] compiler_4.5.1          RSQLite_2.4.5           getPass_0.2-4          
[10] mgcv_1.9-4              png_0.1-8               callr_3.7.6            
[13] vctrs_0.7.1             polylabelr_1.0.0        pkgconfig_2.0.3        
[16] crayon_1.5.3            fastmap_1.2.0           XVector_0.48.0         
[19] labeling_0.4.3          promises_1.5.0          rmarkdown_2.30         
[22] tzdb_0.5.0              UCSC.utils_1.4.0        ps_1.9.1               
[25] ggbeeswarm_0.7.3        bit_4.6.0               xfun_0.56              
[28] cachem_1.1.0            GenomeInfoDb_1.44.3     jsonlite_2.0.0         
[31] blob_1.3.0              later_1.4.5             R6_2.6.1               
[34] bslib_0.10.0            stringi_1.8.7           RColorBrewer_1.1-3     
[37] jquerylib_0.1.4         cellranger_1.1.0        Rcpp_1.1.1             
[40] knitr_1.51              Matrix_1.7-4            httpuv_1.6.16          
[43] splines_4.5.1           timechange_0.4.0        tidyselect_1.2.1       
[46] rstudioapi_0.18.0       yaml_2.3.12             processx_3.8.6         
[49] lattice_0.22-7          withr_3.0.2             KEGGREST_1.48.1        
[52] S7_0.2.1                evaluate_1.0.5          polyclip_1.10-7        
[55] Biostrings_2.76.0       pillar_1.11.1           whisker_0.4.1          
[58] rprojroot_2.1.1         hms_1.1.4               scales_1.4.0           
[61] glue_1.8.0              tools_4.5.1             locfit_1.5-9.12        
[64] fs_1.6.6                grid_4.5.1              Cairo_1.7-0            
[67] nlme_3.1-168            GenomeInfoDbData_1.2.14 beeswarm_0.4.0         
[70] vipor_0.4.7             cli_3.6.5               gtable_0.3.6           
[73] sass_0.4.10             digest_0.6.39           farver_2.1.2           
[76] memoise_2.0.1           htmltools_0.5.9         lifecycle_1.0.5        
[79] httr_1.4.7              statmod_1.5.1           bit64_4.6.0-1          </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
