<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2026-01-14" />

<title>RNA_CorrelationHeatMap</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CrossSpecies_CM_Diff_RNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">RNA_CorrelationHeatMap</h1>
<h4 class="date">2026-01-14</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-01-28
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CrossSpecies_CM_Diff_RNA/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20251129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20251129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20251129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20251129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomJohnDHurleyCrossSpeciesCMDiffRNAtree0fcf7c3f86b6f61814c4e1f7878aa284b8c82266targetblank0fcf7c3a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/tree/0fcf7c3f86b6f61814c4e1f7878aa284b8c82266" target="_blank">0fcf7c3</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomJohnDHurleyCrossSpeciesCMDiffRNAtree0fcf7c3f86b6f61814c4e1f7878aa284b8c82266targetblank0fcf7c3a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/tree/0fcf7c3f86b6f61814c4e1f7878aa284b8c82266" target="_blank">0fcf7c3</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/RNA_CorrelationHeatMap_Ensemble.Rmd</code>) and HTML
(<code>docs/RNA_CorrelationHeatMap_Ensemble.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/0fcf7c3f86b6f61814c4e1f7878aa284b8c82266/analysis/RNA_CorrelationHeatMap_Ensemble.Rmd" target="_blank">0fcf7c3</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
HeatMap knit update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/d9f15a6bf73c125f6d78e861d9c80666a566be7e/docs/RNA_CorrelationHeatMap_Ensemble.html" target="_blank">d9f15a6</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/19cd9fb758bfc6c8a5288ed33632e3a8b6660e6a/analysis/RNA_CorrelationHeatMap_Ensemble.Rmd" target="_blank">19cd9fb</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Commenting out wokrflowr publish
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/3b54f656ea6b3cea32f3c79499a32185c51d97ef/analysis/RNA_CorrelationHeatMap_Ensemble.Rmd" target="_blank">3b54f65</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Duplicate Code Labels
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/de67e6419b2c241e6bb34a1e38406a0f773698c1/analysis/RNA_CorrelationHeatMap_Ensemble.Rmd" target="_blank">de67e64</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Commenting out code not in use.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/da444ed01bf16979957d4ecfedf3a6f72334292f/analysis/RNA_CorrelationHeatMap_Ensemble.Rmd" target="_blank">da444ed</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
RMarkdown name update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/085c1db0f6d1cd1623c082c15b91e46e699b4cc6/analysis/RNA_CorrelationHeatMap_Ensemble.Rmd" target="_blank">085c1db</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Finalizing CorHeatMap
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>####Library Loading####
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;tibble&quot;)
library(&quot;dplyr&quot;)
library(&quot;ggrepel&quot;)
library(&quot;readr&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(&quot;AnnotationDbi&quot;)
library(&quot;pheatmap&quot;)
library(&quot;Cormotif&quot;)
library(&quot;tidyverse&quot;)
library(&quot;workflowr&quot;)
library(&quot;RUVSeq&quot;)
library(&quot;SummarizedExperiment&quot;)
library(&quot;readxl&quot;)
library(&quot;ggfortify&quot;)
library(&quot;ComplexHeatmap&quot;)

# BiocManager::install(&quot;SummarizedExperiment&quot;)

RNA_fc_df &lt;- readRDS(&quot;data/Raw_Data/RNA_fc_df.RDS&quot;)
RNA_Metadata &lt;- readRDS(&quot;data/Raw_Data/RNA_Metadata.RDS&quot;)

RNA_fc &lt;- readRDS(&quot;data/QC/RNA_fc.RDS&quot;)
RNA_log2cpm &lt;- (&quot;data/QC/RNA_log2cpm.RDS&quot;)

Filt_RMG0_RNA_fc &lt;- readRDS(&quot;data/QC/Filt_RMG0_RNA_fc.RDS&quot;)
Filt_RMG0_RNA_log2cpm &lt;-(&quot;data/QC/RNA_log2cpm_RMG0.RDS&quot;)

Cor_Filt_RMG0_RNA_log2cpm &lt;- readRDS(&quot;data/QC/Cor_Filt_RMG0_RNA_log2cpm.RDS&quot;)
Cor_metadata &lt;- readRDS(&quot;data/QC/Cor_RNA_metadata.RDS&quot;)
ann_colors &lt;- readRDS(&quot;data/QC/ann_colors.RDS&quot;)

RNA_Metadata_No4 &lt;- readRDS(&quot;data/QC/RNA_Metatdata_No4.RDS&quot;)
Filt_RMG0_RNA_fc_NoD4 &lt;- readRDS(&quot;data/QC/Filt_RMG0_RNA_fc_NoD4.RDS&quot;)
Filt_RMG0_RNA_log2cpm_NoD4 &lt;- readRDS(&quot;data/QC/Filt_RMG0_RNA_log2cpm_NoD4.RDS&quot;)

Cor_metadata_No4 &lt;- readRDS(&quot;data/QC/Cor_metadata_No4.RDS&quot;)
Cor_Filt_RMG0_RNA_log2cpm_NoD4 &lt;- readRDS(&quot;data/QC/Cor_Filt_RMG0_RNA_log2cpm_NoD4.RDS&quot;)
ann_colors_No4 &lt;- readRDS(&quot;data/QC/ann_colors_no4.RDS&quot;)</code></pre>
<pre class="r"><code># #To keep only OrthoGenes and sample columns
# RNA_fc &lt;- RNA_joined_fc[ , !(names(RNA_joined_fc) %in% c(&quot;gene&quot;,&quot;Chr.x&quot;,&quot;Start.x&quot;,&quot;End.x&quot;,&quot;Strand.x&quot;,&quot;Length.x&quot;,&quot;Geneid.x&quot;,&quot;Chr.y&quot;,&quot;Start.y&quot;,&quot;End.y&quot;,&quot;Strand.y&quot;,&quot;Length.y&quot;,&quot;Geneid.y&quot;))]
# # 89 columns; 7 x 6 = 42 Human Exp, 7 x 6 = 42 Chimp Exp, 4 Human Replicate
# rownames(RNA_fc) &lt;- RNA_joined_fc$gene
# 
# #Rename column Names to More Useful Info
col_names &lt;- c(&quot;H28126_D0&quot;,
              &quot;H28126_D2&quot;,
              &quot;H28126_D4&quot;,
              &quot;H28126_D5&quot;,
              &quot;H28126_D15&quot;,
              &quot;H28126_D30&quot;,
              &quot;H17_D0&quot;,
              &quot;H17_D2&quot;,
              &quot;H17_D4&quot;,
              &quot;H17_D5&quot;,
              &quot;H17_D15&quot;,
              &quot;H17_D30&quot;,
              &quot;H78_D0&quot;,
              &quot;H78_D2&quot;,
              &quot;H78_D4&quot;,
              &quot;H78_D5&quot;,
              &quot;H78_D15&quot;,
              &quot;H78_D30&quot;,
              &quot;H20682_D0&quot;,
              &quot;H20682_D2&quot;,
              &quot;H20682_D4&quot;,
              &quot;H20682_D5&quot;,
              &quot;H20682_D15&quot;,
              &quot;H20682_D30&quot;,
              &quot;H22422_D0&quot;,
              &quot;H22422_D2&quot;,
              &quot;H22422_D4&quot;,
              &quot;H22422_D5&quot;,
              &quot;H22422_D15&quot;,
              &quot;H22422_D30&quot;,
              &quot;H21792_D0&quot;,
              &quot;H21792_D2&quot;,
              &quot;H21792_D4&quot;,
              &quot;H21792_D5&quot;,
              &quot;H21792_D15&quot;,
              &quot;H21792_D30&quot;,
              &quot;H24280_D0&quot;,
              &quot;H24280_D2&quot;,
              &quot;H24280_D4&quot;,
              &quot;H24280_D5&quot;,
              &quot;H24280_D15&quot;,
              &quot;H24280_D30&quot;,
              &quot;H20682R_D0&quot;,
              &quot;H20682R_D2&quot;,
              &quot;H20682R_D5&quot;,
              &quot;H20682R_D30&quot;,
              &quot;C3649_D0&quot;,
              &quot;C3649_D2&quot;,
              &quot;C3649_D4&quot;,
              &quot;C3649_D5&quot;,
              &quot;C3649_D15&quot;,
              &quot;C3649_D30&quot;,
              &quot;C4955_D0&quot;,
              &quot;C4955_D2&quot;,
              &quot;C4955_D4&quot;,
              &quot;C4955_D5&quot;,
              &quot;C4955_D15&quot;,
              &quot;C4955_D30&quot;,
              &quot;C3651_D0&quot;,
              &quot;C3651_D2&quot;,
              &quot;C3651_D4&quot;,
              &quot;C3651_D5&quot;,
              &quot;C3651_D15&quot;,
              &quot;C3651_D30&quot;,
              &quot;C40210_D0&quot;,
              &quot;C40210_D2&quot;,
              &quot;C40210_D4&quot;,
              &quot;C40210_D5&quot;,
              &quot;C40210_D15&quot;,
              &quot;C40210_D30&quot;,
              &quot;C8861_D0&quot;,
              &quot;C8861_D2&quot;,
              &quot;C8861_D4&quot;,
              &quot;C8861_D5&quot;,
              &quot;C8861_D15&quot;,
              &quot;C8861_D30&quot;,
              &quot;C40280_D0&quot;,
              &quot;C40280_D2&quot;,
              &quot;C40280_D4&quot;,
              &quot;C40280_D5&quot;,
              &quot;C40280_D15&quot;,
              &quot;C40280_D30&quot;,
              &quot;C3647_D0&quot;,
              &quot;C3647_D2&quot;,
              &quot;C3647_D4&quot;,
              &quot;C3647_D5&quot;,
              &quot;C3647_D15&quot;,
              &quot;C3647_D30&quot;
)
# colnames(RNA_fc) &lt;- col_names 
# dim(RNA_fc)
# 
# sum(duplicated(rownames(RNA_fc)))
# ensembl_ids_unfilt &lt;- rownames(RNA_fc)
# entrez_ids_unfilt &lt;- mapIds(org.Hs.eg.db,
#                             keys = ensembl_ids_unfilt,
#                             column = &quot;ENTREZID&quot;,
#                             keytype = &quot;ENSEMBL&quot;,
#                             multiVals = &quot;first&quot;)
# symbol_ids_unfilt &lt;- mapIds(org.Hs.eg.db,
#                             keys = ensembl_ids_unfilt,
#                             column = &quot;SYMBOL&quot;,
#                             keytype = &quot;ENSEMBL&quot;,
#                             multiVals = &quot;first&quot;)
# 
# RNA_fc_df &lt;- as.data.frame(RNA_fc)
# RNA_fc_df &lt;- RNA_fc_df %&gt;%
#   rownames_to_column(var = &quot;Ensemble&quot;) %&gt;%
#   dplyr::mutate(
#     Entrez_ID = entrez_ids_unfilt,
#     Symbol    = symbol_ids_unfilt
#   ) %&gt;%
#   dplyr::select(
#     Ensemble,        # 1st column
#     Entrez_ID,       # 2nd column
#     Symbol,          # 3rd column
#     everything()     # rest unchanged
#   )
# 
# # saveRDS(RNA_fc_df,&quot;data/Data_Frames/RNA_fc_df.RDS&quot;)
# 
# RNA_Metadata &lt;- read_excel(&quot;~/diff_timeline_tes/RNA/RNA_Metadata.xlsx&quot;)
# 
# # saveRDS(RNA_Metadata,&quot;data/Data_Frames/RNA_Metadata.RDS&quot;)</code></pre>
<pre class="r"><code>#####Unfiltered####
RNA_fc &lt;- RNA_fc_df %&gt;% 
  dplyr::select(c(-&quot;Entrez_ID&quot;, -&quot;Symbol&quot;)) %&gt;% 
  column_to_rownames(&quot;Ensemble&quot;)

# saveRDS(RNA_fc,&quot;data/QC/RNA_fc.RDS&quot;)

RNA_log2cpm &lt;- cpm(RNA_fc,log=TRUE)
print(hist(RNA_log2cpm,  main = &quot;Histogram of all counts (unfiltered)&quot;,
     xlab =expression(&quot;Log&quot;[2]*&quot; counts-per-million&quot;), col =4 ))</code></pre>
<p><img src="figure/RNA_CorrelationHeatMap_Ensemble.Rmd/RNASeq_FeatureCounts_Box_Unfiltered-1.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNASeq_FeatureCounts_Box_Unfiltered-1">
Past versions of RNASeq_FeatureCounts_Box_Unfiltered-1.png
</button>
</p>
<div id="fig-RNASeq_FeatureCounts_Box_Unfiltered-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/d9f15a6bf73c125f6d78e861d9c80666a566be7e/docs/figure/RNA_CorrelationHeatMap_Ensemble.Rmd/RNASeq_FeatureCounts_Box_Unfiltered-1.png" target="_blank">d9f15a6</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>$breaks
 [1] -3 -2 -1  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16

$counts
 [1] 2026734  332340  206543  159084  134510  130097  148151  195453  219757
[10]  174507   94391   39658   15034    5070    1267     310      70      21
[19]       3

$density
 [1] 5.219506e-01 8.558846e-02 5.319160e-02 4.096935e-02 3.464074e-02
 [6] 3.350425e-02 3.815375e-02 5.033557e-02 5.659464e-02 4.494128e-02
[11] 2.430878e-02 1.021324e-02 3.871749e-03 1.305691e-03 3.262941e-04
[16] 7.983518e-05 1.802730e-05 5.408190e-06 7.725985e-07

$mids
 [1] -2.5 -1.5 -0.5  0.5  1.5  2.5  3.5  4.5  5.5  6.5  7.5  8.5  9.5 10.5 11.5
[16] 12.5 13.5 14.5 15.5

$xname
[1] &quot;RNA_log2cpm&quot;

$equidist
[1] TRUE

attr(,&quot;class&quot;)
[1] &quot;histogram&quot;</code></pre>
<pre class="r"><code>boxplot(RNA_log2cpm, main = &quot;Boxplots of log cpm per sample
          (unfiltered)&quot;, xaxt = &quot;n&quot;, xlab= &quot;&quot;)
axis(1,
     at = 1:length(col_names),  # positions (one per sample)
     labels = col_names,        # your labels vector
     las = 2,               # rotate text vertically (like srt=90)
     cex.axis = 0.3)        # shrink label size</code></pre>
<p><img src="figure/RNA_CorrelationHeatMap_Ensemble.Rmd/RNASeq_FeatureCounts_Box_Unfiltered-2.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNASeq_FeatureCounts_Box_Unfiltered-2">
Past versions of RNASeq_FeatureCounts_Box_Unfiltered-2.png
</button>
</p>
<div id="fig-RNASeq_FeatureCounts_Box_Unfiltered-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/d9f15a6bf73c125f6d78e861d9c80666a566be7e/docs/figure/RNA_CorrelationHeatMap_Ensemble.Rmd/RNASeq_FeatureCounts_Box_Unfiltered-2.png" target="_blank">d9f15a6</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># saveRDS(RNA_log2cpm,&quot;data/QC/RNA_log2cpm.RDS&quot;)</code></pre>
<pre class="r"><code>#####RowMu&gt;0####
row_means &lt;- rowMeans(RNA_log2cpm)
Filt_RMG0_RNA_fc &lt;- RNA_fc[row_means &gt;0,]

# saveRDS(Filt_RMG0_RNA_fc,&quot;data/QC/Filt_RMG0_RNA_fc.RDS&quot;)

Filt_RMG0_RNA_log2cpm &lt;- cpm(Filt_RMG0_RNA_fc,log=TRUE)

# saveRDS(Filt_RMG0_RNA_log2cpm,&quot;data/QC/RNA_log2cpm_RMG0.RDS&quot;)

hist(Filt_RMG0_RNA_log2cpm, main = &quot;Histogram of filtered counts using rowMeans &gt; 0 method&quot;,
     xlab =expression(&quot;Log&quot;[2]*&quot; counts-per-million&quot;), col =5 )</code></pre>
<p><img src="figure/RNA_CorrelationHeatMap_Ensemble.Rmd/RNASeq_FeatureCounts_RMG0-1.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNASeq_FeatureCounts_RMG0-1">
Past versions of RNASeq_FeatureCounts_RMG0-1.png
</button>
</p>
<div id="fig-RNASeq_FeatureCounts_RMG0-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/d9f15a6bf73c125f6d78e861d9c80666a566be7e/docs/figure/RNA_CorrelationHeatMap_Ensemble.Rmd/RNASeq_FeatureCounts_RMG0-1.png" target="_blank">d9f15a6</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>boxplot(Filt_RMG0_RNA_log2cpm, main = &quot;Boxplots of log cpm per sample (RowMeans&gt;0)&quot;,xaxt = &quot;n&quot;, xlab= &quot;&quot;)
axis(1,
     at = 1:length(col_names),  # positions (one per sample)
     labels = col_names,        # your labels vector
     las = 2,               # rotate text vertically (like srt=90)
     cex.axis = 0.3)        # shrink label size</code></pre>
<p><img src="figure/RNA_CorrelationHeatMap_Ensemble.Rmd/RNASeq_FeatureCounts_Box_RMG0-1.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNASeq_FeatureCounts_Box_RMG0-1">
Past versions of RNASeq_FeatureCounts_Box_RMG0-1.png
</button>
</p>
<div id="fig-RNASeq_FeatureCounts_Box_RMG0-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/d9f15a6bf73c125f6d78e861d9c80666a566be7e/docs/figure/RNA_CorrelationHeatMap_Ensemble.Rmd/RNASeq_FeatureCounts_Box_RMG0-1.png" target="_blank">d9f15a6</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>######Cor_HeatMap####
Cor_Filt_RMG0_RNA_log2cpm &lt;- cor(Filt_RMG0_RNA_log2cpm, method = &quot;spearman&quot;)
individual &lt;- RNA_Metadata$Individual
species &lt;- RNA_Metadata$Species
timepoint &lt;- RNA_Metadata$Timepoint
timepoint &lt;- factor(timepoint,levels = c(&quot;Day0&quot;,&quot;Day2&quot;,&quot;Day4&quot;,&quot;Day5&quot;,&quot;Day15&quot;,&quot;Day30&quot;))
Cor_metadata &lt;- data.frame(
  sample_cor = colnames(Filt_RMG0_RNA_log2cpm),
  species_cor = species,
  timepoint_cor = timepoint,
  individual_cor = individual
)


ann_colors &lt;- list(
  timepoint_cor = c(
    &quot;Day0&quot; = &quot;#883268&quot;,   # Purple
    &quot;Day2&quot; = &quot;#3E7274&quot;,  # blue
    &quot;Day4&quot; = &quot;#5AAA464D&quot;,  # light green
    &quot;Day5&quot; = &quot;#94C47D&quot;,  # Green
    &quot;Day15&quot; = &quot;#C03830&quot;,  # red
    &quot;Day30&quot; = &quot;#830C05&quot;  # dark red
  ),
  species_cor = c(
    &quot;H&quot; = &quot;#171717&quot;,  # black
    &quot;C&quot; = &quot;#17171717&quot;   # light grey
  ),
  individual_cor = c(
    H1 = &quot;#091638&quot;, #Blue-Green Darkest
    H2 = &quot;#11185B&quot;,
    H3 = &quot;#0F2C71&quot;,
    H4 = &quot;#0D568F&quot;,
    H4R = &quot;#0D568F&quot;,
    H5 = &quot;#1D8296&quot;,
    H6 = &quot;#46A389&quot;,
    H7 = &quot;#9DD484&quot;, #Blue-Green Lightest
    C1 = &quot;#340702&quot;, #Brown-Orange darkest
    C2 = &quot;#5D0B02&quot;,
    C3 = &quot;#951302&quot;,
    C4 = &quot;#D32804&quot;,
    C5 = &quot;#F74019&quot;,
    C6 = &quot;#FA7A38&quot;,
    C7 = &quot;#FCC598&quot;
    
  )
)
rownames(Cor_metadata) &lt;- Cor_metadata$sample_cor

# saveRDS(Cor_Filt_RMG0_RNA_log2cpm, &quot;data/QC/Cor_Filt_RMG0_RNA_log2cpm.RDS&quot;)
# saveRDS(Cor_metadata, &quot;data/QC/Cor_RNA_metadata.RDS&quot;)
# saveRDS(ann_colors,&quot;data/QC/ann_colors.RDS&quot;)</code></pre>
<pre class="r"><code>print(
  pheatmap(Cor_Filt_RMG0_RNA_log2cpm,
         fontsize_row = 5,
         fontsize_col = 5,
         annotation_col = Cor_metadata[, c(&quot;species_cor&quot;, &quot;timepoint_cor&quot;,&quot;individual_cor&quot;)],
         annotation_colors = ann_colors,
         clustering_distance_rows = &quot;correlation&quot;,
         clustering_distance_cols = &quot;correlation&quot;,
         main = &quot;Sample-Sample Correlation \n(Spearman-log2CPM-RowMeans&gt;0)&quot;)
)</code></pre>
<p><img src="figure/RNA_CorrelationHeatMap_Ensemble.Rmd/RNASeq_FeatureCounts_HeatMap_RMG0-1.png" alt="" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNASeq_FeatureCounts_HeatMap_RMG0-1">
Past versions of RNASeq_FeatureCounts_HeatMap_RMG0-1.png
</button>
</p>
<div id="fig-RNASeq_FeatureCounts_HeatMap_RMG0-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/d9f15a6bf73c125f6d78e861d9c80666a566be7e/docs/figure/RNA_CorrelationHeatMap_Ensemble.Rmd/RNASeq_FeatureCounts_HeatMap_RMG0-1.png" target="_blank">d9f15a6</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>####Subset####
RNA_Metadata_No4 &lt;- RNA_Metadata %&gt;% 
 filter(timepoint != &quot;Day4&quot;)

RNA_fc_NoD4 &lt;- RNA_fc %&gt;% 
  dplyr::select(-ends_with(&quot;_D4&quot;))

RNA_log2cpm_NoD4 &lt;- cpm(RNA_fc_NoD4,log=TRUE)
dim(RNA_log2cpm_NoD4)</code></pre>
<pre><code>[1] 44125    74</code></pre>
<pre class="r"><code>dim(RNA_fc)</code></pre>
<pre><code>[1] 44125    88</code></pre>
<pre class="r"><code>row_means_NoD4 &lt;- rowMeans(RNA_log2cpm_NoD4)
Filt_RMG0_RNA_fc_NoD4 &lt;- RNA_fc_NoD4[row_means_NoD4 &gt;0,]
dim(Filt_RMG0_RNA_fc_NoD4)</code></pre>
<pre><code>[1] 14838    74</code></pre>
<pre class="r"><code>Filt_RMG0_RNA_log2cpm_NoD4 &lt;- cpm(Filt_RMG0_RNA_fc_NoD4,log=TRUE)

# saveRDS(RNA_Metadata_No4,&quot;data/QC/RNA_Metatdata_No4.RDS&quot;)
# saveRDS(Filt_RMG0_RNA_fc_NoD4,&quot;data/QC/Filt_RMG0_RNA_fc_NoD4.RDS&quot;)
# saveRDS(Filt_RMG0_RNA_log2cpm_NoD4, &quot;data/QC/Filt_RMG0_RNA_log2cpm_NoD4.RDS&quot;)</code></pre>
<pre class="r"><code>######Cor_HeatMap####
Cor_Filt_RMG0_RNA_log2cpm_NoD4 &lt;- cor(Filt_RMG0_RNA_log2cpm_NoD4, method = &quot;spearman&quot;)

Cor_metadata_No4 &lt;- Cor_metadata %&gt;% 
  dplyr::filter(timepoint_cor !=&quot;Day4&quot;)

ann_colors_No4 &lt;- ann_colors
ann_colors_No4$timepoint_cor &lt;- ann_colors$timepoint_cor[
  names(ann_colors$timepoint_cor) != &quot;Day4&quot;
]

# saveRDS(Cor_metadata_No4, &quot;data/QC/Cor_metadata_No4.RDS&quot;)
# saveRDS(Cor_Filt_RMG0_RNA_log2cpm_NoD4, &quot;data/QC/Cor_Filt_RMG0_RNA_log2cpm_NoD4.RDS&quot;)
# saveRDS(ann_colors_No4,&quot;data/QC/ann_colors_no4.RDS&quot;)</code></pre>
<pre class="r"><code>print(
  pheatmap(Cor_Filt_RMG0_RNA_log2cpm_NoD4,
         fontsize_row = 5,
         fontsize_col = 5,
         annotation_col = Cor_metadata_No4[, c(&quot;species_cor&quot;, &quot;timepoint_cor&quot;,&quot;individual_cor&quot;)],
         annotation_colors = ann_colors_No4,
         clustering_distance_rows = &quot;correlation&quot;,
         clustering_distance_cols = &quot;correlation&quot;,
         main = &quot;Sample-Sample Correlation (Spearman) \n (log2CPM-RowMeans&gt;0-NoDay4)&quot;)
)</code></pre>
<p><img src="figure/RNA_CorrelationHeatMap_Ensemble.Rmd/rRNASeq_FeatureCounts_HeatMap_RMG0_No4-1.png" alt="" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-rRNASeq_FeatureCounts_HeatMap_RMG0_No4-1">
Past versions of rRNASeq_FeatureCounts_HeatMap_RMG0_No4-1.png
</button>
</p>
<div id="fig-rRNASeq_FeatureCounts_HeatMap_RMG0_No4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/d9f15a6bf73c125f6d78e861d9c80666a566be7e/docs/figure/RNA_CorrelationHeatMap_Ensemble.Rmd/rRNASeq_FeatureCounts_HeatMap_RMG0_No4-1.png" target="_blank">d9f15a6</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>filt_gene_list &lt;- rownames(Filt_RMG0_RNA_log2cpm_NoD4)
#14838
length(filt_gene_list)</code></pre>
<pre><code>[1] 14838</code></pre>
<pre class="r"><code># # in order to make this match with annot later down the line, change the col names for counts_raw_matrix to match final_sample_names in annot
# 
# # i&#39;ll also want to make sure I keep the replicate for this set
# 
# Ind_RUV &lt;- c(rep(&quot;H1&quot;,5),
#              rep(&quot;H2&quot;,5),
#              rep(&quot;H3&quot;,5),
#              rep(&quot;H4&quot;,5),
#              rep(&quot;H5&quot;,5),
#              rep(&quot;H6&quot;,5),
#              rep(&quot;H7&quot;,5),
#              rep(&quot;H4R&quot;,4),
#              rep(&quot;C1&quot;,5),
#              rep(&quot;C2&quot;,5),
#              rep(&quot;C3&quot;,5),
#              rep(&quot;C4&quot;,5),
#              rep(&quot;C5&quot;,5),
#              rep(&quot;C6&quot;,5),
#              rep(&quot;C7&quot;,5)
#              )
# RNA_Metadata_No4$Ind_RUV &lt;- Ind_RUV
# RNA_Metadata_No4_RUV &lt;- RNA_Metadata_No4 %&gt;%
#    mutate(
#     Ind_RUV = factor(Ind_RUV,
#                         levels = c(&quot;H1&quot;,&quot;H2&quot;,&quot;H3&quot;,&quot;H4&quot;,&quot;H5&quot;,&quot;H6&quot;,&quot;H7&quot;,&quot;H4R&quot;,&quot;C1&quot;,&quot;C2&quot;,&quot;C3&quot;,&quot;C4&quot;,&quot;C5&quot;,&quot;C6&quot;,&quot;C7&quot;),ordered=TRUE),
#     Timepoint = factor(Timepoint,
#                        levels = c(&quot;Day0&quot;,&quot;Day2&quot;,&quot;Day5&quot;,&quot;Day15&quot;,&quot;Day30&quot;),ordered=TRUE),
#     Species = factor(Species,
#                      levels = c(&quot;H&quot;,&quot;C&quot;),ordered=TRUE),
#     Condition = factor(
#       Condition,
#       levels = expand.grid(
#         Ind_RUV = levels(Ind_RUV),
#         Timepoint = levels(Timepoint),
#         Species = levels(Species)
#       ) %&gt;%
#         transmute(Condition=paste(Species,Timepoint,Ind_RUV,sep=&quot;_&quot;)) %&gt;%
#         pull(Condition),
#       ordered=TRUE
#     )
#   )
# 
# # saveRDS(RNA_Metadata_No4_RUV,&quot;data/QC/RNA_MetaData_NoD4_RUV.RDS&quot;)
# RNA_Metadata_No4_RUV &lt;- readRDS(&quot;data/QC/RNA_MetaData_NoD4_RUV.RDS&quot;)
# 
# RNA_fc_NoD4_RUV &lt;- RNA_fc_NoD4
# colnames(RNA_fc_NoD4_RUV) &lt;- RNA_Metadata_No4_RUV$Condition
# 
# RUV_filt_counts &lt;- RNA_fc_NoD4_RUV %&gt;% 
#   as.data.frame() %&gt;% 
#   dplyr::filter(., row.names(.)%in% filt_gene_list)
# # saveRDS(RUV_filt_counts, &quot;data/DGE/RUV_filt_counts.RDS&quot;)
# dim(RUV_filt_counts)
# 
# #add in the annotation files
# ind_num &lt;- RNA_Metadata_No4_RUV$Ind_RUV
# type(ind_num)
# length(ind_num)
# 
# annot &lt;- as.data.frame(RNA_Metadata_No4_RUV)
# type(annot)
# is.data.frame(annot)
# #same as Metadata
# 
# 
# #  counts need to be integer values and in a numeric matrix
# # note: the log transformation needs to be accounted for in the isLog argument in RUVs function.
# counts &lt;- as.matrix(RUV_filt_counts)
#   
# # saveRDS(counts, &quot;data/QC/filt_counts_matrix.RDS&quot;)
# 
# # Create a DataFrame for the phenoData
# phenoData &lt;- DataFrame(annot)
# 
# set &lt;- SummarizedExperiment(assays = counts, metadata = phenoData)
# 
# scIdx &lt;- RUVSeq::makeGroups(phenoData$Cond)
# 
# # # Generate a background matrix
# # # The column &quot;Cond&quot; holds the comparisons that you actually want to make. DOX_24, DMSO_24,5FU_24, DOX_3,etc.
# # Day0 &lt;- c(16,36)
# # Day2 &lt;- c(17,37)
# # Day5 &lt;- c(18,38)
# # Day30 &lt;- c(20,39)
# # scIdx &lt;- rbind(Day0,Day2,Day5,Day30)
# # rownames(scIdx) &lt;- c(&quot;[1,]&quot;,&quot;[2,]&quot;,&quot;[3,]&quot;,&quot;[4,]&quot;)
# # colnames(scIdx) &lt;- c(&quot;[,1]&quot;,&quot;[,2]&quot;)
# ```
# 
# 
# 
# ```{r}
# #now I&#39;ve made all of the data I need for this - they are located in each section for k values
# 
# #DO NOT USE THESE COUNTS FOR LINEAR MODELING
# 
# #colors for all of the plots
# # txtime_col
# # ind_col
# # time_col
# # tx_col
# log2cpm &lt;- cpm(counts,log=TRUE)
# # before ruv (counts PCA)
# prcomp_res_log2 &lt;- prcomp(t(Filt_RMG0_RNA_log2cpm_NoD4), scale=FALSE, center = TRUE)
# annot_prcomp_res &lt;- prcomp_res_log2$x %&gt;% cbind(., annot)
# # 
# # group_2 &lt;- annot$dgelist
# # dgelist_col &lt;- annot$dgelist
# # individual_list &lt;- annot$Individual</code></pre>
<pre class="r"><code># 
# #now plot my PCA for filtered counts
# ####PC1/PC2####
# individual_list &lt;- RNA_Metadata_No4_RUV$Individual
# 
# suppressWarnings(
#   ggplot2::autoplot(prcomp_res_log2,
#                     data = annot,
#                     colour = &quot;Timepoint&quot;,
#                     shape = &quot;Species&quot;,
#                     size = 4,
#                     x = 1,
#                     y = 2) +
#     scale_color_manual(values = ann_colors_No4$timepoint_cor) +
#     ggrepel::geom_text_repel(
#       label = individual_list,
#       vjust = -0.5,
#       max.overlaps = 50
#     ) +
#     ggtitle(&quot;RNA log2cpm RMG0&quot;)
# )</code></pre>
<pre class="r"><code># set1 &lt;- RUVSeq::RUVs(x = counts, k =1, scIdx = scIdx, isLog = FALSE)
# 
# # Get the RUV factors (weights) for modeling
# RUV_df1 &lt;- set1$W %&gt;% as.data.frame()
# RUV_df1$Condition &lt;- rownames(RUV_df1)
# 
# RUV_df_rm1 &lt;- RUV_df1[RUV_df1$Condition %in% annot$Condition, ]
# View(RUV_df_rm1)
# RUV_1 &lt;- RUV_df_rm1$W_1
# 
# saveRDS(RUV_df_rm1, &quot;data/QC/RUV_df_rm1.RDS&quot;)
# saveRDS(RUV_1, &quot;data/QC/RUV_1.RDS&quot;)
# 
# 
# log2cpm_k1 &lt;- cpm(set1$normalizedCounts, log = TRUE)
# 
# prcomp_res_log2_k1 &lt;- prcomp(t(log2cpm_k1), scale=FALSE, center = TRUE)
# annot_prcomp_res_k1 &lt;- prcomp_res_log2_k1$x %&gt;% cbind(., RNA_Metadata_No4)
# #PCA checks
# #k=1
# 
# ggplot2::autoplot(prcomp_res_log2_k1,
#                   data = annot_prcomp_res_k1,
#                   colour = &quot;Timepoint&quot;,
#                   shape = &quot;Species&quot;,
#                   size = 4,
#                   x=1,
#                   y=2) +
#   scale_color_manual(values=ann_colors_No4$timepoint_cor)+
#   ggrepel::geom_text_repel(label = individual_list,
#                             vjust = -0.5,
#                             max.overlaps = 50)+
#   ggtitle(&quot;RUV Correction k=1 log2cpm&quot;)</code></pre>
<pre class="r"><code># git -&gt; commit all changes
# git -&gt; push
# wflow_publish(&quot;analysis/RNA_CorrelationHeatMap_Ensemble.Rmd&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.5.1 (2025-06-13 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 26100)

Matrix products: default
  LAPACK version 3.12.1

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ComplexHeatmap_2.24.1       ggfortify_0.4.19           
 [3] readxl_1.4.5                RUVSeq_1.42.0              
 [5] EDASeq_2.42.0               ShortRead_1.66.0           
 [7] GenomicAlignments_1.44.0    SummarizedExperiment_1.38.1
 [9] MatrixGenerics_1.20.0       matrixStats_1.5.0          
[11] Rsamtools_2.24.0            GenomicRanges_1.60.0       
[13] Biostrings_2.76.0           GenomeInfoDb_1.44.3        
[15] XVector_0.48.0              BiocParallel_1.42.1        
[17] lubridate_1.9.4             forcats_1.0.1              
[19] stringr_1.5.2               purrr_1.1.0                
[21] tidyr_1.3.1                 tidyverse_2.0.0            
[23] Cormotif_1.54.0             affy_1.86.0                
[25] pheatmap_1.0.13             org.Hs.eg.db_3.21.0        
[27] AnnotationDbi_1.70.0        IRanges_2.42.0             
[29] S4Vectors_0.46.0            Biobase_2.68.0             
[31] BiocGenerics_0.54.0         generics_0.1.4             
[33] readr_2.1.5                 ggrepel_0.9.6              
[35] dplyr_1.1.4                 tibble_3.3.0               
[37] ggplot2_4.0.0               edgeR_4.6.3                
[39] limma_3.64.3                workflowr_1.7.2            

loaded via a namespace (and not attached):
  [1] later_1.4.4             BiocIO_1.18.0           bitops_1.0-9           
  [4] filelock_1.0.3          R.oo_1.27.1             cellranger_1.1.0       
  [7] preprocessCore_1.70.0   XML_3.99-0.20           lifecycle_1.0.5        
 [10] httr2_1.2.2             pwalign_1.4.0           doParallel_1.0.17      
 [13] rprojroot_2.1.1         processx_3.8.6          lattice_0.22-7         
 [16] MASS_7.3-65             magrittr_2.0.3          sass_0.4.10            
 [19] rmarkdown_2.30          jquerylib_0.1.4         yaml_2.3.10            
 [22] httpuv_1.6.16           otel_0.2.0              DBI_1.2.3              
 [25] RColorBrewer_1.1-3      abind_1.4-8             R.utils_2.13.0         
 [28] RCurl_1.98-1.17         rappdirs_0.3.4          git2r_0.36.2           
 [31] circlize_0.4.17         GenomeInfoDbData_1.2.14 codetools_0.2-20       
 [34] DelayedArray_0.34.1     xml2_1.5.1              tidyselect_1.2.1       
 [37] shape_1.4.6.1           UCSC.utils_1.4.0        farver_2.1.2           
 [40] BiocFileCache_2.16.2    jsonlite_2.0.0          GetoptLong_1.1.0       
 [43] iterators_1.0.14        foreach_1.5.2           tools_4.5.1            
 [46] progress_1.2.3          Rcpp_1.1.0              glue_1.8.0             
 [49] gridExtra_2.3           SparseArray_1.8.1       xfun_0.53              
 [52] withr_3.0.2             BiocManager_1.30.27     fastmap_1.2.0          
 [55] latticeExtra_0.6-31     callr_3.7.6             digest_0.6.37          
 [58] timechange_0.3.0        R6_2.6.1                colorspace_2.1-2       
 [61] Cairo_1.7-0             jpeg_0.1-11             biomaRt_2.64.0         
 [64] RSQLite_2.4.3           R.methodsS3_1.8.2       rtracklayer_1.68.0     
 [67] prettyunits_1.2.0       httr_1.4.7              S4Arrays_1.8.1         
 [70] whisker_0.4.1           pkgconfig_2.0.3         gtable_0.3.6           
 [73] blob_1.3.0              S7_0.2.0                hwriter_1.3.2.1        
 [76] htmltools_0.5.8.1       clue_0.3-66             scales_1.4.0           
 [79] png_0.1-8               knitr_1.51              rstudioapi_0.18.0      
 [82] tzdb_0.5.0              rjson_0.2.23            curl_7.0.0             
 [85] cachem_1.1.0            GlobalOptions_0.1.3     parallel_4.5.1         
 [88] restfulr_0.0.16         pillar_1.11.1           vctrs_0.6.5            
 [91] promises_1.3.3          dbplyr_2.5.1            cluster_2.1.8.1        
 [94] evaluate_1.0.5          GenomicFeatures_1.60.0  cli_3.6.5              
 [97] locfit_1.5-9.12         compiler_4.5.1          rlang_1.1.6            
[100] crayon_1.5.3            interp_1.1-6            aroma.light_3.38.0     
[103] ps_1.9.1                getPass_0.2-4           fs_1.6.6               
[106] stringi_1.8.7           deldir_2.0-4            Matrix_1.7-3           
[109] hms_1.1.4               bit64_4.6.0-1           KEGGREST_1.48.1        
[112] statmod_1.5.0           memoise_2.0.1           affyio_1.78.0          
[115] bslib_0.9.0             bit_4.6.0              </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
