<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2026-01-22" />

<title>RNA_PCA</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CrossSpecies_CM_Diff_RNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">RNA_PCA</h1>
<h4 class="date">2026-01-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-01-28
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CrossSpecies_CM_Diff_RNA/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20251129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20251129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20251129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20251129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomJohnDHurleyCrossSpeciesCMDiffRNAtreea7fc350b3f12b70d6e33a73499c51ea0a0f3b253targetblanka7fc350a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/tree/a7fc350b3f12b70d6e33a73499c51ea0a0f3b253" target="_blank">a7fc350</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomJohnDHurleyCrossSpeciesCMDiffRNAtreea7fc350b3f12b70d6e33a73499c51ea0a0f3b253targetblanka7fc350a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/tree/a7fc350b3f12b70d6e33a73499c51ea0a0f3b253" target="_blank">a7fc350</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/RNA_PCA_Ensemble.Rmd</code>) and
HTML (<code>docs/RNA_PCA_Ensemble.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/a7fc350b3f12b70d6e33a73499c51ea0a0f3b253/analysis/RNA_PCA_Ensemble.Rmd" target="_blank">a7fc350</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
wflowr site update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/6783fc9c030a876700816271f1e6bcdb06464eea/analysis/RNA_PCA_Ensemble.Rmd" target="_blank">6783fc9</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
PCA polishing
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/085c1db0f6d1cd1623c082c15b91e46e699b4cc6/analysis/RNA_PCA_Ensemble.Rmd" target="_blank">085c1db</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Finalizing CorHeatMap
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>####Library Loading####
library(&quot;edgeR&quot;)</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre class="r"><code>library(&quot;ggplot2&quot;)
library(&quot;tibble&quot;)
library(&quot;dplyr&quot;)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(&quot;ggrepel&quot;)
library(&quot;readr&quot;)
library(&quot;org.Hs.eg.db&quot;)</code></pre>
<pre><code>Loading required package: AnnotationDbi</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>Loading required package: generics</code></pre>
<pre><code>
Attaching package: &#39;generics&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    explain</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,
    setequal, union</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    combine</code></pre>
<pre><code>The following object is masked from &#39;package:limma&#39;:

    plotMA</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,
    unsplit, which.max, which.min</code></pre>
<pre><code>Loading required package: Biobase</code></pre>
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    first, rename</code></pre>
<pre><code>The following object is masked from &#39;package:utils&#39;:

    findMatches</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname</code></pre>
<pre><code>
Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    collapse, desc, slice</code></pre>
<pre><code>The following object is masked from &#39;package:grDevices&#39;:

    windows</code></pre>
<pre><code>
Attaching package: &#39;AnnotationDbi&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    select</code></pre>
<pre><code></code></pre>
<pre class="r"><code>library(&quot;AnnotationDbi&quot;)
library(&quot;pheatmap&quot;)
library(&quot;Cormotif&quot;)</code></pre>
<pre><code>Loading required package: affy</code></pre>
<pre class="r"><code>library(&quot;tidyverse&quot;)</code></pre>
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ forcats   1.0.1     ✔ stringr   1.5.2
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.1.0     </code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ lubridate::%within%()    masks IRanges::%within%()
✖ IRanges::collapse()      masks dplyr::collapse()
✖ Biobase::combine()       masks BiocGenerics::combine(), dplyr::combine()
✖ IRanges::desc()          masks dplyr::desc()
✖ tidyr::expand()          masks S4Vectors::expand()
✖ dplyr::filter()          masks stats::filter()
✖ S4Vectors::first()       masks dplyr::first()
✖ dplyr::lag()             masks stats::lag()
✖ lubridate::pm()          masks affy::pm()
✖ BiocGenerics::Position() masks ggplot2::Position(), base::Position()
✖ purrr::reduce()          masks IRanges::reduce()
✖ S4Vectors::rename()      masks dplyr::rename()
✖ lubridate::second()      masks S4Vectors::second()
✖ lubridate::second&lt;-()    masks S4Vectors::second&lt;-()
✖ AnnotationDbi::select()  masks dplyr::select()
✖ IRanges::slice()         masks dplyr::slice()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(&quot;workflowr&quot;)
library(&quot;RUVSeq&quot;)</code></pre>
<pre><code>Loading required package: EDASeq
Loading required package: ShortRead
Loading required package: BiocParallel
Loading required package: Biostrings
Loading required package: XVector

Attaching package: &#39;XVector&#39;

The following object is masked from &#39;package:purrr&#39;:

    compact

Loading required package: GenomeInfoDb

Attaching package: &#39;Biostrings&#39;

The following object is masked from &#39;package:base&#39;:

    strsplit

Loading required package: Rsamtools
Loading required package: GenomicRanges
Loading required package: GenomicAlignments
Loading required package: SummarizedExperiment
Loading required package: MatrixGenerics
Loading required package: matrixStats</code></pre>
<pre><code>Warning: package &#39;matrixStats&#39; was built under R version 4.5.2</code></pre>
<pre><code>
Attaching package: &#39;matrixStats&#39;

The following objects are masked from &#39;package:Biobase&#39;:

    anyMissing, rowMedians

The following object is masked from &#39;package:dplyr&#39;:

    count


Attaching package: &#39;MatrixGenerics&#39;

The following objects are masked from &#39;package:matrixStats&#39;:

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars

The following object is masked from &#39;package:Biobase&#39;:

    rowMedians


Attaching package: &#39;GenomicAlignments&#39;

The following object is masked from &#39;package:dplyr&#39;:

    last


Attaching package: &#39;ShortRead&#39;

The following object is masked from &#39;package:purrr&#39;:

    compose

The following object is masked from &#39;package:affy&#39;:

    intensity

The following object is masked from &#39;package:dplyr&#39;:

    id

The following object is masked from &#39;package:tibble&#39;:

    view</code></pre>
<pre class="r"><code>library(&quot;SummarizedExperiment&quot;)
library(&quot;readxl&quot;)
library(&quot;ggfortify&quot;)</code></pre>
<pre><code>Warning: package &#39;ggfortify&#39; was built under R version 4.5.2</code></pre>
<pre class="r"><code>library(&quot;ComplexHeatmap&quot;)</code></pre>
<pre><code>Loading required package: grid

Attaching package: &#39;grid&#39;

The following object is masked from &#39;package:Biostrings&#39;:

    pattern

========================================
ComplexHeatmap version 2.24.1
Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
Github page: https://github.com/jokergoo/ComplexHeatmap
Documentation: http://jokergoo.github.io/ComplexHeatmap-reference

If you use it in published research, please cite either one:
- Gu, Z. Complex Heatmap Visualization. iMeta 2022.
- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
    genomic data. Bioinformatics 2016.


The new InteractiveComplexHeatmap package can directly export static 
complex heatmaps into an interactive Shiny app with zero effort. Have a try!

This message can be suppressed by:
  suppressPackageStartupMessages(library(ComplexHeatmap))
========================================
! pheatmap() has been masked by ComplexHeatmap::pheatmap(). Most of the arguments
   in the original pheatmap() are identically supported in the new function. You 
   can still use the original function by explicitly calling pheatmap::pheatmap().


Attaching package: &#39;ComplexHeatmap&#39;

The following object is masked from &#39;package:pheatmap&#39;:

    pheatmap</code></pre>
<pre class="r"><code># BiocManager::install(&quot;SummarizedExperiment&quot;)

RNA_fc_df &lt;- readRDS(&quot;data/Raw_Data/RNA_fc_df.RDS&quot;)
RNA_Metadata &lt;- readRDS(&quot;data/Raw_Data/RNA_Metadata.RDS&quot;)

RNA_fc &lt;- readRDS(&quot;data/QC/RNA_fc.RDS&quot;)
RNA_log2cpm &lt;- readRDS(&quot;data/QC/RNA_log2cpm.RDS&quot;)

Filt_RMG0_RNA_fc &lt;- readRDS(&quot;data/QC/Filt_RMG0_RNA_fc.RDS&quot;)
Filt_RMG0_RNA_log2cpm &lt;- readRDS(&quot;data/QC/RNA_log2cpm_RMG0.RDS&quot;)

ann_colors &lt;- readRDS(&quot;data/QC/ann_colors.RDS&quot;)

RNA_Metadata_No4 &lt;- readRDS(&quot;data/QC/RNA_Metatdata_No4.RDS&quot;)
Filt_RMG0_RNA_fc_NoD4 &lt;- readRDS(&quot;data/QC/Filt_RMG0_RNA_fc_NoD4.RDS&quot;)
Filt_RMG0_RNA_log2cpm_NoD4 &lt;- readRDS(&quot;data/QC/Filt_RMG0_RNA_log2cpm_NoD4.RDS&quot;)

ann_colors_No4 &lt;- readRDS(&quot;data/QC/ann_colors_no4.RDS&quot;)

prcomp_RNA &lt;- readRDS(&quot;data/QC/prcomp_RNA.RDS&quot;)
prcomp_RNA_meta &lt;- readRDS(&quot;data/QC/prcomp_RNA_meta.RDS&quot;)
prcomp_Filt_RMG0_RNA &lt;- readRDS(&quot;data/QC/prcomp_Filt_RMG0_RNA.RDS&quot;)
prcomp_Filt_RMG0_RNA_meta &lt;- readRDS(&quot;data/QC/prcomp_Filt_RMG0_RNA_meta.RDS&quot;)
prcomp_Filt_RMG0_RNA_log2cpm_NoD4 &lt;- readRDS(&quot;data/QC/prcomp_Filt_RMG0_RNA_log2cpm_NoD4.RDS&quot;)
prcomp_Filt_RMG0_RNA_log2cpm_NoD4_meta &lt;- readRDS(&quot;data/QC/prcomp_Filt_RMG0_RNA_log2cpm_NoD4_meta.RDS&quot;)</code></pre>
<pre class="r"><code>prcomp_RNA &lt;- prcomp(t(RNA_log2cpm), scale=FALSE, center = TRUE)
prcomp_RNA_meta &lt;- prcomp_RNA$x %&gt;% cbind(., RNA_Metadata)
#PCA checks
#k=1

ggplot2::autoplot(prcomp_RNA,
                  data = prcomp_RNA_meta,
                  colour = &quot;Timepoint&quot;,
                  shape = &quot;Species&quot;,
                  size = 4,
                  x=1,
                  y=2) +
  scale_color_manual(values=ann_colors$timepoint_cor)+
  ggrepel::geom_text_repel(label = RNA_Metadata$Individual,
                            vjust = -0.5,
                            max.overlaps = 50)+
  ggtitle(&quot;Unfiltered RNA Log2CPM PC1-PC2&quot;)</code></pre>
<pre><code>Warning: `aes_string()` was deprecated in ggplot2 3.0.0.
ℹ Please use tidy evaluation idioms with `aes()`.
ℹ See also `vignette(&quot;ggplot2-in-packages&quot;)` for more information.
ℹ The deprecated feature was likely used in the ggfortify package.
  Please report the issue at &lt;https://github.com/sinhrks/ggfortify/issues&gt;.
This warning is displayed once per session.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<p><img src="figure/RNA_PCA_Ensemble.Rmd/RNA_Unfiltered_PCA_PC1_PC2-1.png" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code># saveRDS(prcomp_RNA,&quot;data/QC/prcomp_RNA.RDS&quot;)
# saveRDS(prcomp_RNA_meta,&quot;data/QC/prcomp_RNA_meta.RDS&quot;)</code></pre>
<pre class="r"><code>prcomp_Filt_RMG0_RNA &lt;- prcomp(t(Filt_RMG0_RNA_log2cpm), scale=FALSE, center = TRUE)
prcomp_Filt_RMG0_RNA_meta &lt;- prcomp_Filt_RMG0_RNA$x %&gt;% cbind(., RNA_Metadata)
#PCA checks
#k=1

ggplot2::autoplot(prcomp_Filt_RMG0_RNA ,
                  data = prcomp_Filt_RMG0_RNA_meta,
                  colour = &quot;Timepoint&quot;,
                  shape = &quot;Species&quot;,
                  size = 4,
                  x=1,
                  y=2) +
  scale_color_manual(values=ann_colors$timepoint_cor)+
  ggrepel::geom_text_repel(label = RNA_Metadata$Individual,
                            vjust = -0.5,
                            max.overlaps = 50)+
  ggtitle(&quot;Filtered RNA Log2CPM RMG0&quot;)</code></pre>
<p><img src="figure/RNA_PCA_Ensemble.Rmd/RNA_RMG0_PCA_PC1_PC2-1.png" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code># saveRDS(prcomp_Filt_RMG0_RNA,&quot;data/QC/prcomp_Filt_RMG0_RNA.RDS&quot;)
# saveRDS(prcomp_Filt_RMG0_RNA_meta,&quot;data/QC/prcomp_Filt_RMG0_RNA_meta.RDS&quot;)</code></pre>
<pre class="r"><code>prcomp_Filt_RMG0_RNA_log2cpm_NoD4 &lt;- prcomp(t(Filt_RMG0_RNA_log2cpm_NoD4), scale=FALSE, center = TRUE)
prcomp_Filt_RMG0_RNA_log2cpm_NoD4_meta &lt;- prcomp_Filt_RMG0_RNA_log2cpm_NoD4$x %&gt;% cbind(., RNA_Metadata_No4)
#PCA checks
#k=1

ggplot2::autoplot(prcomp_Filt_RMG0_RNA_log2cpm_NoD4 ,
                  data = prcomp_Filt_RMG0_RNA_log2cpm_NoD4_meta,
                  colour = &quot;Timepoint&quot;,
                  shape = &quot;Species&quot;,
                  size = 4,
                  x=1,
                  y=2) +
  scale_color_manual(values=ann_colors_No4$timepoint_cor)+
  ggrepel::geom_text_repel(label = RNA_Metadata_No4$Individual,
                            vjust = -0.5,
                            max.overlaps = 50)+
  ggtitle(&quot;Filtered RNA Log2CPM RMG0 No Day 4&quot;)</code></pre>
<p><img src="figure/RNA_PCA_Ensemble.Rmd/RNA_NoD4_RMG0_PCA_PC1_PC2-1.png" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code># saveRDS(prcomp_Filt_RMG0_RNA_log2cpm_NoD4,&quot;data/QC/prcomp_Filt_RMG0_RNA_log2cpm_NoD4.RDS&quot;)
# saveRDS(prcomp_Filt_RMG0_RNA_log2cpm_NoD4_meta,&quot;data/QC/prcomp_Filt_RMG0_RNA_log2cpm_NoD4_meta.RDS&quot;)</code></pre>
<pre class="r"><code>ggplot2::autoplot(prcomp_Filt_RMG0_RNA_log2cpm_NoD4 ,
                  data = prcomp_Filt_RMG0_RNA_log2cpm_NoD4_meta,
                  colour = &quot;Timepoint&quot;,
                  shape = &quot;Species&quot;,
                  size = 4,
                  x=3,
                  y=4) +
  scale_color_manual(values=ann_colors_No4$timepoint_cor)+
  ggrepel::geom_text_repel(label = RNA_Metadata_No4$Individual,
                            vjust = -0.5,
                            max.overlaps = 50)+
  ggtitle(&quot;Filtered RNA Log2CPM RMG0 No Day 4&quot;)</code></pre>
<p><img src="figure/RNA_PCA_Ensemble.Rmd/RNA_NoD4_RMG0_PCA_PC3_PC4-1.png" alt="" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code># saveRDS(prcomp_Filt_RMG0_RNA_log2cpm_NoD4,&quot;data/QC/prcomp_Filt_RMG0_RNA_log2cpm_NoD4.RDS&quot;)
# saveRDS(prcomp_Filt_RMG0_RNA_log2cpm_NoD4_meta,&quot;data/QC/prcomp_Filt_RMG0_RNA_log2cpm_NoD4_meta.RDS&quot;)</code></pre>
<pre class="r"><code># git -&gt; commit all changes
# git -&gt; push
# wflow_publish(&quot;analysis/RNA_PCA_Ensemble.Rmd&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.5.1 (2025-06-13 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 26100)

Matrix products: default
  LAPACK version 3.12.1

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ComplexHeatmap_2.24.1       ggfortify_0.4.19           
 [3] readxl_1.4.5                RUVSeq_1.42.0              
 [5] EDASeq_2.42.0               ShortRead_1.66.0           
 [7] GenomicAlignments_1.44.0    SummarizedExperiment_1.38.1
 [9] MatrixGenerics_1.20.0       matrixStats_1.5.0          
[11] Rsamtools_2.24.0            GenomicRanges_1.60.0       
[13] Biostrings_2.76.0           GenomeInfoDb_1.44.3        
[15] XVector_0.48.0              BiocParallel_1.42.1        
[17] lubridate_1.9.4             forcats_1.0.1              
[19] stringr_1.5.2               purrr_1.1.0                
[21] tidyr_1.3.1                 tidyverse_2.0.0            
[23] Cormotif_1.54.0             affy_1.86.0                
[25] pheatmap_1.0.13             org.Hs.eg.db_3.21.0        
[27] AnnotationDbi_1.70.0        IRanges_2.42.0             
[29] S4Vectors_0.46.0            Biobase_2.68.0             
[31] BiocGenerics_0.54.0         generics_0.1.4             
[33] readr_2.1.5                 ggrepel_0.9.6              
[35] dplyr_1.1.4                 tibble_3.3.0               
[37] ggplot2_4.0.0               edgeR_4.6.3                
[39] limma_3.64.3                workflowr_1.7.2            

loaded via a namespace (and not attached):
  [1] later_1.4.4             BiocIO_1.18.0           bitops_1.0-9           
  [4] filelock_1.0.3          R.oo_1.27.1             cellranger_1.1.0       
  [7] preprocessCore_1.70.0   XML_3.99-0.20           lifecycle_1.0.5        
 [10] httr2_1.2.2             pwalign_1.4.0           doParallel_1.0.17      
 [13] rprojroot_2.1.1         processx_3.8.6          lattice_0.22-7         
 [16] MASS_7.3-65             magrittr_2.0.3          sass_0.4.10            
 [19] rmarkdown_2.30          jquerylib_0.1.4         yaml_2.3.10            
 [22] httpuv_1.6.16           otel_0.2.0              DBI_1.2.3              
 [25] RColorBrewer_1.1-3      abind_1.4-8             R.utils_2.13.0         
 [28] RCurl_1.98-1.17         rappdirs_0.3.4          git2r_0.36.2           
 [31] circlize_0.4.17         GenomeInfoDbData_1.2.14 codetools_0.2-20       
 [34] DelayedArray_0.34.1     xml2_1.5.1              tidyselect_1.2.1       
 [37] shape_1.4.6.1           UCSC.utils_1.4.0        farver_2.1.2           
 [40] BiocFileCache_2.16.2    jsonlite_2.0.0          GetoptLong_1.1.0       
 [43] iterators_1.0.14        foreach_1.5.2           tools_4.5.1            
 [46] progress_1.2.3          Rcpp_1.1.0              glue_1.8.0             
 [49] gridExtra_2.3           SparseArray_1.8.1       xfun_0.53              
 [52] withr_3.0.2             BiocManager_1.30.27     fastmap_1.2.0          
 [55] latticeExtra_0.6-31     callr_3.7.6             digest_0.6.37          
 [58] timechange_0.3.0        R6_2.6.1                colorspace_2.1-2       
 [61] jpeg_0.1-11             biomaRt_2.64.0          RSQLite_2.4.3          
 [64] R.methodsS3_1.8.2       rtracklayer_1.68.0      prettyunits_1.2.0      
 [67] httr_1.4.7              S4Arrays_1.8.1          whisker_0.4.1          
 [70] pkgconfig_2.0.3         gtable_0.3.6            blob_1.3.0             
 [73] S7_0.2.0                hwriter_1.3.2.1         htmltools_0.5.8.1      
 [76] clue_0.3-66             scales_1.4.0            png_0.1-8              
 [79] knitr_1.51              rstudioapi_0.18.0       tzdb_0.5.0             
 [82] rjson_0.2.23            curl_7.0.0              cachem_1.1.0           
 [85] GlobalOptions_0.1.3     parallel_4.5.1          restfulr_0.0.16        
 [88] pillar_1.11.1           vctrs_0.6.5             promises_1.3.3         
 [91] dbplyr_2.5.1            cluster_2.1.8.1         evaluate_1.0.5         
 [94] GenomicFeatures_1.60.0  cli_3.6.5               locfit_1.5-9.12        
 [97] compiler_4.5.1          rlang_1.1.6             crayon_1.5.3           
[100] labeling_0.4.3          interp_1.1-6            aroma.light_3.38.0     
[103] ps_1.9.1                getPass_0.2-4           fs_1.6.6               
[106] stringi_1.8.7           deldir_2.0-4            Matrix_1.7-3           
[109] hms_1.1.4               bit64_4.6.0-1           KEGGREST_1.48.1        
[112] statmod_1.5.0           memoise_2.0.1           affyio_1.78.0          
[115] bslib_0.9.0             bit_4.6.0              </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
