<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2026-01-28" />

<title>RNA_CHDGene_Comp_Species</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CrossSpecies_CM_Diff_RNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">RNA_CHDGene_Comp_Species</h1>
<h4 class="date">2026-01-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-01-28
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CrossSpecies_CM_Diff_RNA/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20251129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20251129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20251129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20251129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomJohnDHurleyCrossSpeciesCMDiffRNAtree3686961e37e4f1fdefb145a1f5c391aaebde8359targetblank3686961a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/tree/3686961e37e4f1fdefb145a1f5c391aaebde8359" target="_blank">3686961</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomJohnDHurleyCrossSpeciesCMDiffRNAtree3686961e37e4f1fdefb145a1f5c391aaebde8359targetblank3686961a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/tree/3686961e37e4f1fdefb145a1f5c391aaebde8359" target="_blank">3686961</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/RNA_CHDGene_Comp_Species.Rmd</code>) and HTML
(<code>docs/RNA_CHDGene_Comp_Species.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/3686961e37e4f1fdefb145a1f5c391aaebde8359/analysis/RNA_CHDGene_Comp_Species.Rmd" target="_blank">3686961</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
publish change
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/582a67da607d5935a2b0fa52257e568494de27ee/analysis/RNA_CHDGene_Comp_Species.Rmd" target="_blank">582a67d</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Code block headers added
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/5a82fc9a8d71b8b86965cc3eceff773f92a0e2b9/analysis/RNA_CHDGene_Comp_Species.Rmd" target="_blank">5a82fc9</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
CHDGene website
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/085c1db0f6d1cd1623c082c15b91e46e699b4cc6/analysis/RNA_CHDGene_Comp_Species.Rmd" target="_blank">085c1db</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Finalizing CorHeatMap
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>#Loading Libraries
library(dplyr)
library(tidyr)
library(stringr)
library(ggplot2)
library(biomaRt)
library(tidyverse)
library(readxl)
library(readr)
library(reshape2)
library(circlize)
library(grid)
library(stringr)
library(org.Hs.eg.db)
library(AnnotationDbi)
library(gprofiler2)
library(purrr)
# install.packages(&quot;biomartr&quot;)

#Loading R Objects
# quadrant_gene_lists &lt;- readRDS(&quot;data/DGE/quadrant_gene_lists_DayBefore.RDS&quot;)
# chdgene_table &lt;- readRDS(&quot;data/DGE/chdgene_table.RDS&quot;)
Day0_DEGs_HC &lt;- readRDS(&quot;data/DGE/Species/Day0_DEGs_HC.RDS&quot;)
Day2_DEGs_HC &lt;- readRDS(&quot;data/DGE/Species/Day2_DEGs_HC.RDS&quot;)
Day5_DEGs_HC &lt;- readRDS(&quot;data/DGE/Species/Day5_DEGs_HC.RDS&quot;)
Day15_DEGs_HC &lt;- readRDS(&quot;data/DGE/Species/Day15_DEGs_HC.RDS&quot;)
Day30_DEGs_HC &lt;- readRDS(&quot;data/DGE/Species/Day30_DEGs_HC.RDS&quot;)
all_intersections &lt;- readRDS(&quot;data/DGE/Species/all_intersections.RDS&quot;)</code></pre>
<pre class="r"><code>generate_volcano_plot &lt;- function(toptable, title) {
  
  # #check for entrezid 
  # if(!&quot;Entrez_ID&quot; %in% colnames(toptable)) stop(&quot;Entrez_ID col not present&quot;)
  # 
  #make significance labels
  
  toptable &lt;- toptable %&gt;% 
    mutate(Significance = case_when(
      logFC &gt; 0 &amp; adj.P.Val &lt; 0.05 ~ &quot;Upregulated&quot;,
      logFC &lt; 0 &amp; adj.P.Val &lt; 0.05 ~ &quot;Downregulated&quot;,
      TRUE  ~ &quot;Not Significant&quot;
    ))
  
  #factor significance
  toptable$Significance &lt;- factor(
    toptable$Significance, 
    levels = c(&quot;Upregulated&quot;, 
               &quot;Not Significant&quot;, 
               &quot;Downregulated&quot;)
  )
  
  #count genes in each category
  upgenes &lt;- sum(toptable$Significance == &quot;Upregulated&quot;)
  nsgenes &lt;- sum(toptable$Significance == &quot;Not Significant&quot;)
  downgenes &lt;- sum(toptable$Significance == &quot;Downregulated&quot;)
  
  
  #labels for legend
  legend_lab &lt;- c(
    paste0(&quot;Upregulated: &quot;, upgenes),
    paste0(&quot;Not Significant: &quot;, nsgenes),
    paste0(&quot;Downregulated: &quot;, downgenes)
  )
  
  #colors 
  color_map &lt;- c(&quot;Upregulated&quot; = &quot;blue&quot;,
                 &quot;Not Significant&quot; = &quot;grey30&quot;,
                 &quot;Downregulated&quot; = &quot;red&quot;)
  
  #generate volcano plots
  p &lt;- ggplot(toptable, aes(x = logFC, 
                            y = -log10(P.Value),
                            color = Significance)) +
    geom_point_rast(alpha = 0.8, size = 3) +
    scale_color_manual(values = color_map, 
                       labels = legend_lab,
                       breaks = c(&quot;Upregulated&quot;,
                                  &quot;Not Significant&quot;, 
                                  &quot;Downregulated&quot;))  +
    xlim(-20,20) +
    labs(title = title, 
         x = expression(&quot;log&quot;[2]*&quot;FC&quot;),
         y = expression(&quot;-log&quot;[10]*&quot;p-value&quot;)) +
    # theme_custom() +
    theme(legend.position = &quot;right&quot;)
  
  return(p)
}</code></pre>
<pre class="r"><code>generate_volcano_plot_CHDGene &lt;- function(toptable, title, chdgene_genes = NULL) {
  
  # Determine significance
  toptable &lt;- toptable %&gt;% 
    mutate(Significance = case_when(
      logFC &gt; 0 &amp; adj.P.Val &lt; 0.05 ~ &quot;Upregulated&quot;,
      logFC &lt; 0 &amp; adj.P.Val &lt; 0.05 ~ &quot;Downregulated&quot;,
      TRUE  ~ &quot;Not Significant&quot;
    ))

  toptable$Significance &lt;- factor(
    toptable$Significance, 
    levels = c(&quot;Upregulated&quot;, &quot;Not Significant&quot;, &quot;Downregulated&quot;)
  )

  # Mark GWAS hits
  toptable &lt;- toptable %&gt;%
    mutate(highlight = if_else(Gene %in% chdgene_genes, TRUE, FALSE))

  # Split data for layers
  df_non_chdgene &lt;- toptable %&gt;% filter(!highlight)
  df_chdgene     &lt;- toptable %&gt;% filter(highlight)

  # Colors for significance
  color_map &lt;- c(
    &quot;Upregulated&quot; = &quot;blue&quot;,
    &quot;Not Significant&quot; = &quot;grey30&quot;,
    &quot;Downregulated&quot; = &quot;red&quot;
  )

  p &lt;- ggplot() +
    # Non-chdgene points
    geom_point(data = df_non_chdgene,
               aes(x = logFC, y = -log10(P.Value), color = Significance),
               alpha = 0.8, size = 3) +
    scale_color_manual(values = color_map) +

    # GWAS points on top with outline
    geom_point(data = df_chdgene,
               aes(x = logFC, y = -log10(P.Value), fill = Significance),
               shape = 21, color = &quot;limegreen&quot;, size = 4, stroke = 1.5) +

    # Optional: label chdgene genes
    # geom_text_repel(data = df_gwas,
    #                 aes(x = logFC, y = -log10(P.Value), label = Gene),
    #                 size = 3, fontface = &quot;bold&quot;, color = &quot;black&quot;, max.overlaps = 20) +

    xlim(-20, 20) +
    labs(title = title, 
         x = expression(&quot;log&quot;[2]*&quot;FC&quot;),
         y = expression(&quot;-log&quot;[10]*&quot;p-value&quot;)) +
    theme_bw() +
    theme(legend.position = &quot;right&quot;) +
    guides(color = guide_legend(override.aes = list(size = 3)),
           fill = guide_legend(override.aes = list(size = 3)))

  return(p)
}</code></pre>
<pre class="r"><code>chdgene_table &lt;- read.csv(&quot;C:/Users/jdhurley/Downloads/chdgene_table.csv&quot;)
saveRDS(chdgene_table,&quot;data/DGE/chdgene_table.RDS&quot;)

# Make a copy to be safe
chdgene_table_copy &lt;- chdgene_table

# Check
colnames(chdgene_table_copy)</code></pre>
<pre><code>[1] &quot;Gene&quot;                    &quot;CHD.classification&quot;     
[3] &quot;Extra.cardiac.phenotype&quot; &quot;Inheritance.mode&quot;       
[5] &quot;Ranking&quot;                 &quot;Supporting.References&quot;  </code></pre>
<pre class="r"><code># Should include &quot;Gene&quot;

# dim(chdgene_table_copy)
# length(unique(chdgene_table_copy$Gene))
# unique(chdgene_table_copy$Gene)

# Connect to Ensembl
ensembl &lt;- useMart(&quot;ensembl&quot;, dataset = &quot;hsapiens_gene_ensembl&quot;)

# Your Day0-only DEGs (Ensembl IDs)
day0_only &lt;- all_intersections$Day0

# Map Ensembl IDs to gene symbols
mapping &lt;- getBM(
  attributes = c(&quot;ensembl_gene_id&quot;, &quot;hgnc_symbol&quot;),
  filters = &quot;ensembl_gene_id&quot;,
  values = day0_only,
  mart = ensembl
)

# Remove empty mappings
mapping &lt;- mapping[mapping$hgnc_symbol != &quot;&quot;, ]

# Filter by your known gene list
known_genes &lt;- chdgene_table_copy$Gene 
day0_filtered &lt;- mapping$ensembl_gene_id[mapping$hgnc_symbol %in% known_genes]

# Optional: get gene symbols for filtered genes
day0_filtered_symbols &lt;- mapping$hgnc_symbol[mapping$hgnc_symbol %in% known_genes]

saveRDS(day0_filtered_symbols,&quot;data/DGE/Species/Day0_Only_DEGs_CHDGenes.RDS&quot;)

# Your day2-only DEGs (Ensembl IDs)
day2_only &lt;- all_intersections$Day2

# Map Ensembl IDs to gene symbols
mapping &lt;- getBM(
  attributes = c(&quot;ensembl_gene_id&quot;, &quot;hgnc_symbol&quot;),
  filters = &quot;ensembl_gene_id&quot;,
  values = day2_only,
  mart = ensembl
)

# Remove empty mappings
mapping &lt;- mapping[mapping$hgnc_symbol != &quot;&quot;, ]

# Filter by your known gene list
known_genes &lt;- chdgene_table_copy$Gene 
day2_filtered &lt;- mapping$ensembl_gene_id[mapping$hgnc_symbol %in% known_genes]

# Optional: get gene symbols for filtered genes
day2_filtered_symbols &lt;- mapping$hgnc_symbol[mapping$hgnc_symbol %in% known_genes]

saveRDS(day2_filtered_symbols,&quot;data/DGE/Species/Day2_Only_DEGs_CHDGenes.RDS&quot;)

# Your day2-only DEGs (Ensembl IDs)
day5_only &lt;- all_intersections$Day5

# Map Ensembl IDs to gene symbols
mapping &lt;- getBM(
  attributes = c(&quot;ensembl_gene_id&quot;, &quot;hgnc_symbol&quot;),
  filters = &quot;ensembl_gene_id&quot;,
  values = day5_only,
  mart = ensembl
)

# Remove empty mappings
mapping &lt;- mapping[mapping$hgnc_symbol != &quot;&quot;, ]

# Filter by your known gene list
known_genes &lt;- chdgene_table_copy$Gene 
day5_filtered &lt;- mapping$ensembl_gene_id[mapping$hgnc_symbol %in% known_genes]

# Optional: get gene symbols for filtered genes
day5_filtered_symbols &lt;- mapping$hgnc_symbol[mapping$hgnc_symbol %in% known_genes]

saveRDS(day5_filtered_symbols,&quot;data/DGE/Species/day5_Only_DEGs_CHDGenes.RDS&quot;)  

# Your day2-only DEGs (Ensembl IDs)
day15_only &lt;- all_intersections$Day15

# Map Ensembl IDs to gene symbols
mapping &lt;- getBM(
  attributes = c(&quot;ensembl_gene_id&quot;, &quot;hgnc_symbol&quot;),
  filters = &quot;ensembl_gene_id&quot;,
  values = day15_only,
  mart = ensembl
)

# Remove empty mappings
mapping &lt;- mapping[mapping$hgnc_symbol != &quot;&quot;, ]

# Filter by your known gene list
known_genes &lt;- chdgene_table_copy$Gene 
day15_filtered &lt;- mapping$ensembl_gene_id[mapping$hgnc_symbol %in% known_genes]

# Optional: get gene symbols for filtered genes
day15_filtered_symbols &lt;- mapping$hgnc_symbol[mapping$hgnc_symbol %in% known_genes]

saveRDS(day15_filtered_symbols,&quot;data/DGE/Species/day15_Only_DEGs_CHDGenes.RDS&quot;)  

# Your day2-only DEGs (Ensembl IDs)
day30_only &lt;- all_intersections$Day30

# Map Ensembl IDs to gene symbols
mapping &lt;- getBM(
  attributes = c(&quot;ensembl_gene_id&quot;, &quot;hgnc_symbol&quot;),
  filters = &quot;ensembl_gene_id&quot;,
  values = day30_only,
  mart = ensembl
)

# Remove empty mappings
mapping &lt;- mapping[mapping$hgnc_symbol != &quot;&quot;, ]

# Filter by your known gene list
known_genes &lt;- chdgene_table_copy$Gene 
day30_filtered &lt;- mapping$ensembl_gene_id[mapping$hgnc_symbol %in% known_genes]

# Optional: get gene symbols for filtered genes
day30_filtered_symbols &lt;- mapping$hgnc_symbol[mapping$hgnc_symbol %in% known_genes]

saveRDS(day30_filtered_symbols,&quot;data/DGE/Species/day30_Only_DEGs_CHDGenes.RDS&quot;)  

# Your day2-only DEGs (Ensembl IDs)
AllDays_only &lt;- all_intersections$`Day0&amp;Day2&amp;Day5&amp;Day15&amp;Day30`

# Map Ensembl IDs to gene symbols
mapping &lt;- getBM(
  attributes = c(&quot;ensembl_gene_id&quot;, &quot;hgnc_symbol&quot;),
  filters = &quot;ensembl_gene_id&quot;,
  values = AllDays_only,
  mart = ensembl
)

# Remove empty mappings
mapping &lt;- mapping[mapping$hgnc_symbol != &quot;&quot;, ]

# Filter by your known gene list
known_genes &lt;- chdgene_table_copy$Gene 
AllDays_filtered &lt;- mapping$ensembl_gene_id[mapping$hgnc_symbol %in% known_genes]

# Optional: get gene symbols for filtered genes
AllDays_filtered_symbols &lt;- mapping$hgnc_symbol[mapping$hgnc_symbol %in% known_genes]

saveRDS(AllDays_filtered_symbols,&quot;data/DGE/Species/AllDays_Only_DEGs_CHDGenes.RDS&quot;)  </code></pre>
<pre class="r"><code># Create the table
Day0_Only_DEGs_ChiSq &lt;- matrix(c(16, 1719, 8, 1184), nrow = 2, byrow = TRUE)
rownames(Day0_Only_DEGs_ChiSq) &lt;- c(&quot;Day0&quot;, &quot;Day0-2-5-15-30&quot;)
colnames(Day0_Only_DEGs_ChiSq) &lt;- c(&quot;CHDGene&quot;, &quot;Non-CHDGene&quot;)

# Run chi-square test
chisq.test(Day0_Only_DEGs_ChiSq)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  Day0_Only_DEGs_ChiSq
X-squared = 0.2824, df = 1, p-value = 0.5951</code></pre>
<pre class="r"><code># Create the table
Day2_Only_DEGs_ChiSq &lt;- matrix(c(6, 441, 8, 1184), nrow = 2, byrow = TRUE)
rownames(Day2_Only_DEGs_ChiSq) &lt;- c(&quot;Day2&quot;, &quot;Day0-2-5-15-30&quot;)
colnames(Day2_Only_DEGs_ChiSq) &lt;- c(&quot;CHDGene&quot;, &quot;Non-CHDGene&quot;)

# Run chi-square test
chisq.test(Day2_Only_DEGs_ChiSq)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  Day2_Only_DEGs_ChiSq
X-squared = 1.0274, df = 1, p-value = 0.3108</code></pre>
<pre class="r"><code># Create the table
Day5_Only_DEGs_ChiSq &lt;- matrix(c(6,499, 8, 1184), nrow = 2, byrow = TRUE)
rownames(Day5_Only_DEGs_ChiSq) &lt;- c(&quot;Day5&quot;, &quot;Day0-2-5-15-30&quot;)
colnames(Day5_Only_DEGs_ChiSq) &lt;- c(&quot;CHDGene&quot;, &quot;Non-CHDGene&quot;)

# Run chi-square test
chisq.test(Day5_Only_DEGs_ChiSq)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  Day5_Only_DEGs_ChiSq
X-squared = 0.613, df = 1, p-value = 0.4337</code></pre>
<pre class="r"><code># Create the table
Day15_Only_DEGs_ChiSq &lt;- matrix(c(14,602, 8, 1184), nrow = 2, byrow = TRUE)
rownames(Day15_Only_DEGs_ChiSq) &lt;- c(&quot;Day15&quot;, &quot;Day0-2-5-15-30&quot;)
colnames(Day15_Only_DEGs_ChiSq) &lt;- c(&quot;CHDGene&quot;, &quot;Non-CHDGene&quot;)

# Run chi-square test
chisq.test(Day15_Only_DEGs_ChiSq)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  Day15_Only_DEGs_ChiSq
X-squared = 7.3855, df = 1, p-value = 0.006575</code></pre>
<pre class="r"><code># Create the table
Day30_Only_DEGs_ChiSq &lt;- matrix(c(18,1217, 8, 1184), nrow = 2, byrow = TRUE)
rownames(Day30_Only_DEGs_ChiSq) &lt;- c(&quot;Day30&quot;, &quot;Day0-2-5-15-30&quot;)
colnames(Day30_Only_DEGs_ChiSq) &lt;- c(&quot;CHDGene&quot;, &quot;Non-CHDGene&quot;)

# Run chi-square test
chisq.test(Day30_Only_DEGs_ChiSq)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  Day30_Only_DEGs_ChiSq
X-squared = 2.8359, df = 1, p-value = 0.09218</code></pre>
<pre class="r"><code>df &lt;- data.frame(
  Day = c(&quot;Day0&quot;, &quot;Day2&quot;, &quot;Day5&quot;, &quot;Day15&quot;, &quot;Day30&quot;,&quot;Shared_All&quot;),
  CHDGene = c(16,6,6,14,18,8),
  Non_CHDGene = c(1719,441,499,602,1217,1884)
)

# Convert to long format for ggplot
df_long &lt;- df %&gt;%
  pivot_longer(cols = c(CHDGene, Non_CHDGene),
               names_to = &quot;Type&quot;,
               values_to = &quot;Count&quot;) %&gt;%
  group_by(Day) %&gt;%
  mutate(Proportion = Count / sum(Count))  # calculate proportions

df_long$Day &lt;- factor(df_long$Day, levels = c(&quot;Day0&quot;, &quot;Day2&quot;, &quot;Day5&quot;, &quot;Day15&quot;, &quot;Day30&quot;,&quot;Shared_All&quot;), ordered = TRUE)

ggplot(df_long, aes(x = Day, y = Proportion, fill = Type)) +
  geom_bar(stat = &quot;identity&quot;) +
  scale_y_continuous(labels = scales::percent_format()) +  # show % on y-axis
  scale_fill_manual(values = c(&quot;DEGs&quot; = &quot;#E41A1C&quot;, &quot;CHDGene&quot; = &quot;#377EB8&quot;)) + # custom colors
  labs(y = &quot;Proportion&quot;, x = &quot;Day&quot;, fill = &quot;&quot;) +
  theme_minimal(base_size = 14)</code></pre>
<p><img src="figure/RNA_CHDGene_Comp_Species.Rmd/RNA_CHDGene_Species_Barplot-1.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># git -&gt; commit all changes
# git -&gt; push
# wflow_publish(&quot;analysis/RNA_CHDGene_Comp_Species.Rmd&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.5.1 (2025-06-13 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 26100)

Matrix products: default
  LAPACK version 3.12.1

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] gprofiler2_0.2.4     org.Hs.eg.db_3.21.0  AnnotationDbi_1.70.0
 [4] IRanges_2.42.0       S4Vectors_0.46.0     Biobase_2.68.0      
 [7] BiocGenerics_0.54.0  generics_0.1.4       circlize_0.4.17     
[10] reshape2_1.4.5       readxl_1.4.5         lubridate_1.9.4     
[13] forcats_1.0.1        purrr_1.1.0          readr_2.1.5         
[16] tibble_3.3.0         tidyverse_2.0.0      biomaRt_2.64.0      
[19] ggplot2_4.0.0        stringr_1.5.2        tidyr_1.3.1         
[22] dplyr_1.1.4          workflowr_1.7.2     

loaded via a namespace (and not attached):
 [1] DBI_1.2.3               httr2_1.2.2             rlang_1.1.6            
 [4] magrittr_2.0.3          git2r_0.36.2            otel_0.2.0             
 [7] compiler_4.5.1          RSQLite_2.4.3           getPass_0.2-4          
[10] png_0.1-8               callr_3.7.6             vctrs_0.6.5            
[13] pkgconfig_2.0.3         shape_1.4.6.1           crayon_1.5.3           
[16] fastmap_1.2.0           dbplyr_2.5.1            XVector_0.48.0         
[19] labeling_0.4.3          promises_1.3.3          rmarkdown_2.30         
[22] tzdb_0.5.0              UCSC.utils_1.4.0        ps_1.9.1               
[25] bit_4.6.0               xfun_0.53               cachem_1.1.0           
[28] GenomeInfoDb_1.44.3     jsonlite_2.0.0          progress_1.2.3         
[31] blob_1.3.0              later_1.4.4             prettyunits_1.2.0      
[34] R6_2.6.1                bslib_0.9.0             stringi_1.8.7          
[37] RColorBrewer_1.1-3      jquerylib_0.1.4         cellranger_1.1.0       
[40] Rcpp_1.1.0              knitr_1.51              httpuv_1.6.16          
[43] timechange_0.3.0        tidyselect_1.2.1        rstudioapi_0.18.0      
[46] yaml_2.3.10             curl_7.0.0              processx_3.8.6         
[49] plyr_1.8.9              withr_3.0.2             KEGGREST_1.48.1        
[52] S7_0.2.0                evaluate_1.0.5          BiocFileCache_2.16.2   
[55] xml2_1.5.1              Biostrings_2.76.0       pillar_1.11.1          
[58] filelock_1.0.3          whisker_0.4.1           plotly_4.11.0          
[61] rprojroot_2.1.1         hms_1.1.4               scales_1.4.0           
[64] glue_1.8.0              lazyeval_0.2.2          tools_4.5.1            
[67] data.table_1.18.0       fs_1.6.6                colorspace_2.1-2       
[70] GenomeInfoDbData_1.2.14 cli_3.6.5               rappdirs_0.3.4         
[73] viridisLite_0.4.2       gtable_0.3.6            sass_0.4.10            
[76] digest_0.6.37           htmlwidgets_1.6.4       farver_2.1.2           
[79] memoise_2.0.1           htmltools_0.5.8.1       lifecycle_1.0.5        
[82] httr_1.4.7              GlobalOptions_0.1.3     bit64_4.6.0-1          </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
