<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2026-01-28" />

<title>RNA_Cormotif_Comp_Species_Ensembl</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CrossSpecies_CM_Diff_RNA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">RNA_Cormotif_Comp_Species_Ensembl</h1>
<h4 class="date">2026-01-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-02-18
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CrossSpecies_CM_Diff_RNA/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20251129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20251129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20251129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20251129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomJohnDHurleyCrossSpeciesCMDiffRNAtree8e924754032460873644b3da3fe00877d8f672c7targetblank8e92475a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/tree/8e924754032460873644b3da3fe00877d8f672c7" target="_blank">8e92475</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomJohnDHurleyCrossSpeciesCMDiffRNAtree8e924754032460873644b3da3fe00877d8f672c7targetblank8e92475a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/tree/8e924754032460873644b3da3fe00877d8f672c7" target="_blank">8e92475</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    data/

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/RNA_Cormotif_Comp_Species_Ensembl.Rmd</code>) and HTML
(<code>docs/RNA_Cormotif_Comp_Species_Ensembl.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/8e924754032460873644b3da3fe00877d8f672c7/analysis/RNA_Cormotif_Comp_Species_Ensembl.Rmd" target="_blank">8e92475</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-02-18
</td>
<td>
Adding functions from DGE
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/ad16973f37d2c132ff835f4cdbe8537d0c97533d/analysis/RNA_Cormotif_Comp_Species_Ensembl.Rmd" target="_blank">ad16973</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-02-18
</td>
<td>
error in include statment
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/d5937405f1f7825231fe56421364947447ea44e8/analysis/RNA_Cormotif_Comp_Species_Ensembl.Rmd" target="_blank">d593740</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-02-18
</td>
<td>
Typo
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/1d6a7ee3c54be08b833a03d7f7c7c974535b1946/analysis/RNA_Cormotif_Comp_Species_Ensembl.Rmd" target="_blank">1d6a7ee</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-02-18
</td>
<td>
Moving Clust FC to CorMotif
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/b8dfde38c2c2747806f735301e69936058b2589f/analysis/RNA_Cormotif_Comp_Species_Ensembl.Rmd" target="_blank">b8dfde3</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-02-18
</td>
<td>
MovedClustersToCormotif
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/c85837b9cf4737e7032493f7577a8de43816a5ca/analysis/RNA_Cormotif_Comp_Species_Ensembl.Rmd" target="_blank">c85837b</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-02-18
</td>
<td>
SiteUpDate_CorrelationHeatMap
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/3956def95a5c139bb54acf09696f001e7b5a9d03/docs/RNA_Cormotif_Comp_Species_Ensembl.html" target="_blank">3956def</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/07967c8e308b583b4d4f248104b07c6320dad43a/analysis/RNA_Cormotif_Comp_Species_Ensembl.Rmd" target="_blank">07967c8</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Needed to run Cormotif code
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/e6624d7bf2382b57e8c1679f13bc900157e011c4/analysis/RNA_Cormotif_Comp_Species_Ensembl.Rmd" target="_blank">e6624d7</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Duplicate Block names
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/6b3f4af40eb9cd64dcf1a0c0dbec00b84c404149/analysis/RNA_Cormotif_Comp_Species_Ensembl.Rmd" target="_blank">6b3f4af</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
website edit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/6c3cce3528b84cd2a91f8252b3f40ed4706edfbe/analysis/RNA_Cormotif_Comp_Species_Ensembl.Rmd" target="_blank">6c3cce3</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
<td>
Cormotif Species
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE,warning=FALSE,message=FALSE, dev=c(&quot;png&quot;,&quot;pdf&quot;))</code></pre>
<pre class="r"><code>####Library Loading####
library(&quot;edgeR&quot;)
library(&quot;ggplot2&quot;)
library(&quot;tibble&quot;)
library(&quot;dplyr&quot;)
library(&quot;ggrepel&quot;)
library(&quot;readr&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(&quot;AnnotationDbi&quot;)
library(&quot;pheatmap&quot;)
library(&quot;tidyverse&quot;)
library(&quot;workflowr&quot;)
library(&quot;RUVSeq&quot;)
library(&quot;SummarizedExperiment&quot;)
library(&quot;readxl&quot;)
library(&quot;ggfortify&quot;)
library(&quot;ComplexHeatmap&quot;)

# BiocManager::install(&quot;SummarizedExperiment&quot;)

RNA_fc_df &lt;- readRDS(&quot;data/Raw_Data/RNA_fc_df.RDS&quot;)
RNA_Metadata &lt;- readRDS(&quot;data/Raw_Data/RNA_Metadata.RDS&quot;)
ann_colors &lt;- readRDS(&quot;data/QC/ann_colors.RDS&quot;)

RNA_Metadata_No4 &lt;- readRDS(&quot;data/QC/RNA_Metatdata_No4.RDS&quot;)
Filt_RMG0_RNA_fc_NoD4 &lt;- readRDS(&quot;data/QC/Filt_RMG0_RNA_fc_NoD4.RDS&quot;)
Filt_RMG0_RNA_log2cpm_NoD4 &lt;- readRDS(&quot;data/QC/Filt_RMG0_RNA_log2cpm_NoD4.RDS&quot;)
ann_colors_No4 &lt;- readRDS(&quot;data/QC/ann_colors_no4.RDS&quot;)

Filt_RMG0_RNA_fc_NoD4_NoRep &lt;- readRDS(&quot;data/Cormotif/Filt_RMG0_RNA_fc_NoD4_NoRep.RDS&quot;)
Filt_RMG0_RNA_log2cpm_NoD4_NoRep &lt;- readRDS(&quot;data/Cormotif/Filt_RMG0_RNA_log2cpm_NoD4_NoRep.RDS&quot;)

TopTable_RNA_All &lt;- readRDS(&quot;data/TopTable_RNA_all.RDS&quot;)

clust1_RNA &lt;- readRDS(&quot;data/Cormotif/Species/clust1_RNA_Species.RDS&quot;)
clust2_RNA &lt;- readRDS(&quot;data/Cormotif/Species/clust2_RNA_Species.RDS&quot;)
clust3_RNA &lt;- readRDS(&quot;data/Cormotif/Species/clust3_RNA_Species.RDS&quot;)
clust4_RNA &lt;- readRDS(&quot;data/Cormotif/Species/clust4_RNA_Species.RDS&quot;)</code></pre>
<pre class="r"><code>check_overlap &lt;- function(list_of_lists) {
  # create empty list to store results
  overlap_results &lt;- list()
  
  # loop over each cluster
  for (i in seq_along(list_of_lists)) {
    cluster_name &lt;- names(list_of_lists)[i]
    genes_i &lt;- list_of_lists[[i]]
    
    # compare to all other clusters
    overlaps &lt;- sapply(seq_along(list_of_lists), function(j) {
      if (i == j) return(NA)  # skip self
      genes_j &lt;- list_of_lists[[j]]
      length(intersect(genes_i, genes_j))  # count of overlapping genes
    })
    names(overlaps) &lt;- names(list_of_lists)
    
    overlap_results[[cluster_name]] &lt;- overlaps
  }
  
  return(overlap_results)
}</code></pre>
<pre class="r"><code>Filt_RMG0_RNA_log2cpm_NoD4_NoRep &lt;- Filt_RMG0_RNA_log2cpm_NoD4 %&gt;%
  as.data.frame() %&gt;%             # convert to data frame
  dplyr::select(-contains(&quot;R&quot;)) %&gt;%  # remove columns containing &quot;R&quot;
  as.matrix()                     # convert back to matrix

RNA_Metadata_NoD4_NoRep &lt;- RNA_Metadata_No4 %&gt;%
  dplyr::filter(!grepl(&quot;R&quot;, Individual))

# saveRDS(Filt_RMG0_RNA_log2cpm_NoD4_NoRep,&quot;data/Cormotif/Filt_RMG0_RNA_log2cpm_NoD4_NoRep.RDS&quot;)
# saveRDS(RNA_Metadata_NoD4_NoRep,&quot;data/Cormotif/RNA_Metadata_NoD4_NoRep.RDS&quot;)</code></pre>
<pre class="r"><code># groupid &lt;- c(rep(1:5,7),rep(6:10,7))
# compid_Cond1 &lt;- c(1,2,3,4,5)
# compid_Cond2 &lt;- c(6,7,8,9,10)
# compid &lt;- cbind(compid_Cond1,compid_Cond2)
# 
# #Model Fitting
# 
# set.seed(8)
# motif.fitted&lt;-cormotiffit(exprs = Filt_RMG0_RNA_log2cpm_NoD4_NoRep,
#                           groupid = groupid,
#                           compid = compid,
#                           K=1:10,
#                           max.iter = 3000,
#                           runtype = &quot;logCPM&quot;)
# 
# 
# saveRDS(motif.fitted,&quot;data/Cormotif/Species/cormotif_RNA_seed8_k1_10_HC_Day_emmaCode.RDS&quot;)

motif.fitted &lt;- readRDS(&quot;data/Cormotif/Species/cormotif_RNA_seed8_k1_10_HC_Day_emmaCode.RDS&quot;)
plotIC(motif.fitted)</code></pre>
<p><img src="figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_Modeling-1.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNA-Seq_CorMotif_Species_Modeling-1">
Past versions of RNA-Seq_CorMotif_Species_Modeling-1.png
</button>
</p>
<div id="fig-RNA-Seq_CorMotif_Species_Modeling-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/3956def95a5c139bb54acf09696f001e7b5a9d03/docs/figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_Modeling-1.png" target="_blank">3956def</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>motif.fitted$bic</code></pre>
<pre><code>       K      bic
 [1,]  1 455689.5
 [2,]  2 445999.6
 [3,]  3 445588.4
 [4,]  4 445371.9
 [5,]  5 445371.9
 [6,]  6 445427.1
 [7,]  7 445485.9
 [8,]  8 445542.1
 [9,]  9 445601.4
[10,] 10 445659.0</code></pre>
<pre class="r"><code>plotMotif(motif.fitted)</code></pre>
<p><img src="figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_Modeling-2.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNA-Seq_CorMotif_Species_Modeling-2">
Past versions of RNA-Seq_CorMotif_Species_Modeling-2.png
</button>
</p>
<div id="fig-RNA-Seq_CorMotif_Species_Modeling-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/3956def95a5c139bb54acf09696f001e7b5a9d03/docs/figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_Modeling-2.png" target="_blank">3956def</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#  Studies Human Day0 - 1. HDay2, 2. HDay5, 3. HDay15, 4. HDay30
#          Chimp Day0 - 5. CDay2, 6. CDay5, 7. CDay15, 8. CDay30</code></pre>
<pre class="r"><code># Set up Variables
groupid &lt;- c(rep(1:5,7),rep(6:10,7))
compid_Cond1 &lt;- c(1,2,3,4,5)
compid_Cond2 &lt;- c(6,7,8,9,10)
compid &lt;- cbind(compid_Cond1,compid_Cond2)

# Extract motif matrix
motif_matrix &lt;- as.matrix(motif.fitted$bestmotif$motif.q)

# Label rows and columns
rownames(motif_matrix) &lt;- paste0(&quot;Motif_&quot;, seq_len(nrow(motif_matrix)))
colnames(motif_matrix) &lt;- paste0(&quot;C&quot;, compid[,1], &quot;_vs_C&quot;, compid[,2])

# Convert to tidy format
motif_df &lt;- as.data.frame(motif_matrix) %&gt;%
  mutate(Motif = rownames(.)) %&gt;%
  pivot_longer(-Motif, names_to = &quot;Comparison&quot;, values_to = &quot;Probability&quot;)

# Reorder motifs 1–11
motif_df$Motif &lt;- factor(motif_df$Motif, levels = paste0(&quot;Motif_&quot;, 1:4))

motif_df$Comparison &lt;- factor(motif_df$Comparison, levels = c(&quot;C1_vs_C6&quot;,&quot;C2_vs_C7&quot;,&quot;C3_vs_C8&quot;,&quot;C4_vs_C9&quot;,&quot;C5_vs_C10&quot;))

# Plot
ggplot(motif_df, aes(x = Comparison, y = Motif, fill = Probability)) +
  geom_tile(color = &quot;white&quot;) +
  scale_fill_gradient(low = &quot;white&quot;, high = &quot;black&quot;) +
  labs(
    title = &quot;CorMotif Motif Activity Across Comparisons \n  (no day4) (no replicates) (RowMeans&gt;0) Days Across Species&quot; ,
    x = &quot;Comparison (Condition 1 vs Condition 2)&quot;,
    y = &quot;Motif (1–4)&quot;,
    fill = &quot;Posterior Probability&quot;
  ) +
  theme_minimal(base_size = 14) +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    panel.grid = element_blank()
  )</code></pre>
<p><img src="figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_Plotting-1.png" alt="" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNA-Seq_CorMotif_Species_Plotting-1">
Past versions of RNA-Seq_CorMotif_Species_Plotting-1.png
</button>
</p>
<div id="fig-RNA-Seq_CorMotif_Species_Plotting-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/3956def95a5c139bb54acf09696f001e7b5a9d03/docs/figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_Plotting-1.png" target="_blank">3956def</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#  Comparison C1 = Human Day0
#  Comparison C2 = Human Day2
#  Comparison C3 = Human Day5
#  Comparison C4 = Human Day15
#  Comparison C5 = Human Day30
#  Comparison C6 = Chimp Day0
#  Comparison C7 = Chimp Day2
#  Comparison C8 = Chimp Day5
#  Comparison C9 = Chimp Day15
#  Comparison C10 = Chimp Day30</code></pre>
<pre class="r"><code>motif.fit &lt;- motif.fitted
log2cpm &lt;- Filt_RMG0_RNA_log2cpm_NoD4

#extract the posterior probability that these genes belong to motifs
gene_prob_ppost &lt;- motif.fit$bestmotif$p.post
rownames(gene_prob_ppost) &lt;- rownames(log2cpm)
# saveRDS(gene_prob_ppost,&quot;data/Cormotif/Species/RNA_gene_prob_ppost_Species.RDS&quot;)

#assign each gene to a motif with max post prob
assigned_motifs &lt;- apply(gene_prob_ppost, 1, which.max)
max_probs &lt;- apply(gene_prob_ppost, 1, max)

#combine these into a dataframe - motif assigned genes (p.post)
motif_assignment_df &lt;- gene_prob_ppost %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Gene&quot;) %&gt;%
  mutate(
    Assigned_Motif = assigned_motifs[Gene],
    Max_Probability = max_probs[Gene]
  )

#make some histograms of the unfiltered data from Cormotif p.post

gene_prob_ppost %&gt;%
  as.data.frame() %&gt;%
  ggplot(., aes(x = V1))+
  geom_histogram(bins = 50)+
  xlim(0,1)+
  ggtitle(&quot;Cormtif p.post RNA V1&quot;)</code></pre>
<p><img src="figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_Assigning-1.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNA-Seq_CorMotif_Species_Assigning-1">
Past versions of RNA-Seq_CorMotif_Species_Assigning-1.png
</button>
</p>
<div id="fig-RNA-Seq_CorMotif_Species_Assigning-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/3956def95a5c139bb54acf09696f001e7b5a9d03/docs/figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_Assigning-1.png" target="_blank">3956def</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>gene_prob_ppost %&gt;%
  as.data.frame() %&gt;%
  ggplot(., aes(x = V2))+
  geom_histogram(bins = 50)+
  xlim(0,1)+
  ggtitle(&quot;Cormtif p.post RNA V2&quot;)</code></pre>
<p><img src="figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_Assigning-2.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNA-Seq_CorMotif_Species_Assigning-2">
Past versions of RNA-Seq_CorMotif_Species_Assigning-2.png
</button>
</p>
<div id="fig-RNA-Seq_CorMotif_Species_Assigning-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/3956def95a5c139bb54acf09696f001e7b5a9d03/docs/figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_Assigning-2.png" target="_blank">3956def</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>gene_prob_ppost %&gt;%
  as.data.frame() %&gt;%
  ggplot(., aes(x = V3))+
  geom_histogram(bins = 50)+
  xlim(0,1)+
  ggtitle(&quot;Cormtif p.post RNA V3&quot;)</code></pre>
<p><img src="figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_Assigning-3.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNA-Seq_CorMotif_Species_Assigning-3">
Past versions of RNA-Seq_CorMotif_Species_Assigning-3.png
</button>
</p>
<div id="fig-RNA-Seq_CorMotif_Species_Assigning-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/3956def95a5c139bb54acf09696f001e7b5a9d03/docs/figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_Assigning-3.png" target="_blank">3956def</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>gene_prob_ppost %&gt;%
  as.data.frame() %&gt;%
  ggplot(., aes(x = V4))+
  geom_histogram(bins = 50)+
  xlim(0,1)+
  ggtitle(&quot;Cormtif p.post RNA V4&quot;)</code></pre>
<p><img src="figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_Assigning-4.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNA-Seq_CorMotif_Species_Assigning-4">
Past versions of RNA-Seq_CorMotif_Species_Assigning-4.png
</button>
</p>
<div id="fig-RNA-Seq_CorMotif_Species_Assigning-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/3956def95a5c139bb54acf09696f001e7b5a9d03/docs/figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_Assigning-4.png" target="_blank">3956def</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>gene_prob_ppost %&gt;%
  as.data.frame() %&gt;%
  ggplot(., aes(x = V5))+
  geom_histogram(bins = 50)+
  xlim(0,1)+
  ggtitle(&quot;Cormtif p.post RNA V5&quot;)</code></pre>
<p><img src="figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_Assigning-5.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RNA-Seq_CorMotif_Species_Assigning-5">
Past versions of RNA-Seq_CorMotif_Species_Assigning-5.png
</button>
</p>
<div id="fig-RNA-Seq_CorMotif_Species_Assigning-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/JohnDHurley/CrossSpecies_CM_Diff_RNA/blob/3956def95a5c139bb54acf09696f001e7b5a9d03/docs/figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_Assigning-5.png" target="_blank">3956def</a>
</td>
<td>
John D. Hurley
</td>
<td>
2026-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>motif.fit &lt;- motif.fitted
log2cpm &lt;- Filt_RMG0_RNA_log2cpm_NoD4
#extract the cluster likelihood - which DEGs are most likely to be in this cluster
motif_prob &lt;- motif.fit$bestmotif$clustlike
rownames(motif_prob) &lt;- rownames(gene_prob_ppost)
# saveRDS(motif_prob,&quot;data/RNA_motif_prob_Trajectory.RDS&quot;)

# motif.fitted$bestmotif$clustlike

clust1_RNA &lt;-
  motif_prob %&gt;%
  as.data.frame() %&gt;%
  filter(V1 &gt; 0.5 &amp; V2 &lt;  0.5 &amp; V3 &lt; 0.5   &amp; V4 &lt; 0.5)  %&gt;%
  rownames()
print(&quot;clust1_RNA&quot;)</code></pre>
<pre><code>[1] &quot;clust1_RNA&quot;</code></pre>
<pre class="r"><code>length(clust1_RNA)</code></pre>
<pre><code>[1] 10473</code></pre>
<pre class="r"><code>clust2_RNA &lt;-
  motif_prob %&gt;%
  as.data.frame() %&gt;%
  filter(V1 &lt; 0.5 &amp; V2 &gt; 0.5 &amp; V3 &lt; 0.5 &amp; V4 &lt; 0.5)  %&gt;%
  rownames()
print(&quot;clust2_RNA&quot;)</code></pre>
<pre><code>[1] &quot;clust2_RNA&quot;</code></pre>
<pre class="r"><code>length(clust2_RNA)</code></pre>
<pre><code>[1] 525</code></pre>
<pre class="r"><code>clust3_RNA &lt;-
  motif_prob %&gt;%
  as.data.frame() %&gt;%
  filter(V1 &lt; 0.5 &amp; V2 &lt; 0.5 &amp; V3 &gt; 0.5 &amp; V4)  %&gt;%
  rownames()
print(&quot;clust3_RNA&quot;)</code></pre>
<pre><code>[1] &quot;clust3_RNA&quot;</code></pre>
<pre class="r"><code>length(clust3_RNA)</code></pre>
<pre><code>[1] 2129</code></pre>
<pre class="r"><code>clust4_RNA &lt;-
  motif_prob %&gt;%
  as.data.frame() %&gt;%
  filter(V1 &lt; 0.5 &amp; V2 &lt; 0.5 &amp; V3 &lt; 0.5 &amp; V4 &gt; 0.5) %&gt;%
  rownames()
print(&quot;clust4_RNA&quot;)</code></pre>
<pre><code>[1] &quot;clust4_RNA&quot;</code></pre>
<pre class="r"><code>length(clust4_RNA)</code></pre>
<pre><code>[1] 933</code></pre>
<pre class="r"><code>clust_names_list &lt;- list(
  clust1_RNA,
  clust2_RNA,
  clust3_RNA,
  clust4_RNA
)

names(clust_names_list) &lt;- paste0(&quot;clust&quot;, 1:4)

print(overlap_matirx &lt;- check_overlap(clust_names_list))</code></pre>
<pre><code>$clust1
clust1 clust2 clust3 clust4 
    NA      0      0      0 

$clust2
clust1 clust2 clust3 clust4 
     0     NA      0      0 

$clust3
clust1 clust2 clust3 clust4 
     0      0     NA      0 

$clust4
clust1 clust2 clust3 clust4 
     0      0      0     NA </code></pre>
<pre class="r"><code>TopTable_RNA_All$AbsFC &lt;- abs(TopTable_RNA_All$logFC)
cluster_names &lt;- c(&quot;clust1_RNA&quot;,&quot;clust2_RNA&quot;,&quot;clust3_RNA&quot;,&quot;clust4_RNA&quot;)</code></pre>
<pre class="r"><code>cluster_lists &lt;- list(
  clust1_RNA  = clust1_RNA,
  clust2_RNA  = clust2_RNA,
  clust3_RNA  = clust3_RNA,
  clust4_RNA  = clust4_RNA
)

cluster_plots &lt;- lapply(names(cluster_lists), function(clust_name) {
  plot_cluster_fc(cluster_lists[[clust_name]], clust_name, TopTable_RNA_All)
})</code></pre>
<pre><code>clust1_RNA : 52365 rows, 10473 unique genes
clust2_RNA : 2625 rows, 525 unique genes
clust3_RNA : 10645 rows, 2129 unique genes
clust4_RNA : 4665 rows, 933 unique genes</code></pre>
<pre class="r"><code>names(cluster_plots) &lt;- names(cluster_lists)

for (p in cluster_plots) print(p)</code></pre>
<p><img src="figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_ClustSets_DGE_FC-1.png" alt="" width="672" style="display: block; margin: auto;" /><img src="figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_ClustSets_DGE_FC-2.png" alt="" width="672" style="display: block; margin: auto;" /><img src="figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_ClustSets_DGE_FC-3.png" alt="" width="672" style="display: block; margin: auto;" /><img src="figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_ClustSets_DGE_FC-4.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cluster_lists &lt;- list(
  clust1_RNA  = clust1_RNA,
  clust2_RNA  = clust2_RNA,
  clust3_RNA  = clust3_RNA,
  clust4_RNA  = clust4_RNA
)

cluster_plots &lt;- lapply(names(cluster_lists), function(clust_name) {
  plot_cluster_absfc(cluster_lists[[clust_name]], clust_name, TopTable_RNA_All)
})</code></pre>
<pre><code>clust1_RNA : 52365 rows, 10473 unique genes
clust2_RNA : 2625 rows, 525 unique genes
clust3_RNA : 10645 rows, 2129 unique genes
clust4_RNA : 4665 rows, 933 unique genes</code></pre>
<pre class="r"><code>names(cluster_plots) &lt;- names(cluster_lists)

for (p in cluster_plots) print(p)</code></pre>
<p><img src="figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_ClustSets_DGE_AbsFC-1.png" alt="" width="672" style="display: block; margin: auto;" /><img src="figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_ClustSets_DGE_AbsFC-2.png" alt="" width="672" style="display: block; margin: auto;" /><img src="figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_ClustSets_DGE_AbsFC-3.png" alt="" width="672" style="display: block; margin: auto;" /><img src="figure/RNA_Cormotif_Comp_Species_Ensembl.Rmd/RNA-Seq_CorMotif_Species_ClustSets_DGE_AbsFC-4.png" alt="" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># git -&gt; commit all changes
# git -&gt; push
# wflow_publish(&quot;analysis/RNA_Cormotif_Comp_Species_Ensembl.Rmd&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.5.1 (2025-06-13 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 26100)

Matrix products: default
  LAPACK version 3.12.1

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ComplexHeatmap_2.24.1       ggfortify_0.4.19           
 [3] readxl_1.4.5                RUVSeq_1.42.0              
 [5] EDASeq_2.42.0               ShortRead_1.66.0           
 [7] GenomicAlignments_1.44.0    SummarizedExperiment_1.38.1
 [9] MatrixGenerics_1.20.0       matrixStats_1.5.0          
[11] Rsamtools_2.24.0            GenomicRanges_1.60.0       
[13] Biostrings_2.76.0           GenomeInfoDb_1.44.3        
[15] XVector_0.48.0              BiocParallel_1.42.1        
[17] lubridate_1.9.5             forcats_1.0.1              
[19] stringr_1.6.0               purrr_1.2.1                
[21] tidyr_1.3.2                 tidyverse_2.0.0            
[23] pheatmap_1.0.13             org.Hs.eg.db_3.21.0        
[25] AnnotationDbi_1.70.0        IRanges_2.42.0             
[27] S4Vectors_0.46.0            Biobase_2.68.0             
[29] BiocGenerics_0.54.1         generics_0.1.4             
[31] readr_2.1.6                 ggrepel_0.9.6              
[33] dplyr_1.1.4                 tibble_3.3.1               
[35] ggplot2_4.0.2               edgeR_4.6.3                
[37] limma_3.64.3                workflowr_1.7.2            

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3      shape_1.4.6.1           rstudioapi_0.18.0      
  [4] jsonlite_2.0.0          magrittr_2.0.4          GenomicFeatures_1.60.0 
  [7] farver_2.1.2            rmarkdown_2.30          GlobalOptions_0.1.3    
 [10] BiocIO_1.18.0           fs_1.6.6                vctrs_0.7.1            
 [13] memoise_2.0.1           RCurl_1.98-1.17         htmltools_0.5.9        
 [16] S4Arrays_1.8.1          progress_1.2.3          curl_7.0.0             
 [19] cellranger_1.1.0        SparseArray_1.8.1       sass_0.4.10            
 [22] bslib_0.10.0            httr2_1.2.2             cachem_1.1.0           
 [25] whisker_0.4.1           iterators_1.0.14        lifecycle_1.0.5        
 [28] pkgconfig_2.0.3         Matrix_1.7-4            R6_2.6.1               
 [31] fastmap_1.2.0           clue_0.3-66             GenomeInfoDbData_1.2.14
 [34] digest_0.6.39           colorspace_2.1-2        ps_1.9.1               
 [37] rprojroot_2.1.1         RSQLite_2.4.5           hwriter_1.3.2.1        
 [40] labeling_0.4.3          filelock_1.0.3          timechange_0.4.0       
 [43] httr_1.4.7              abind_1.4-8             compiler_4.5.1         
 [46] doParallel_1.0.17       bit64_4.6.0-1           withr_3.0.2            
 [49] S7_0.2.1                DBI_1.2.3               R.utils_2.13.0         
 [52] biomaRt_2.64.0          MASS_7.3-65             rappdirs_0.3.4         
 [55] DelayedArray_0.34.1     rjson_0.2.23            tools_4.5.1            
 [58] otel_0.2.0              httpuv_1.6.16           R.oo_1.27.1            
 [61] glue_1.8.0              restfulr_0.0.16         callr_3.7.6            
 [64] promises_1.5.0          getPass_0.2-4           cluster_2.1.8.1        
 [67] gtable_0.3.6            tzdb_0.5.0              R.methodsS3_1.8.2      
 [70] hms_1.1.4               xml2_1.5.2              foreach_1.5.2          
 [73] pillar_1.11.1           later_1.4.5             circlize_0.4.17        
 [76] BiocFileCache_2.16.2    lattice_0.22-7          aroma.light_3.38.0     
 [79] rtracklayer_1.68.0      bit_4.6.0               deldir_2.0-4           
 [82] tidyselect_1.2.1        locfit_1.5-9.12         knitr_1.51             
 [85] git2r_0.36.2            gridExtra_2.3           xfun_0.56              
 [88] statmod_1.5.1           stringi_1.8.7           UCSC.utils_1.4.0       
 [91] yaml_2.3.12             evaluate_1.0.5          codetools_0.2-20       
 [94] interp_1.1-6            cli_3.6.5               processx_3.8.6         
 [97] jquerylib_0.1.4         Rcpp_1.1.1              dbplyr_2.5.1           
[100] png_0.1-8               XML_3.99-0.20           parallel_4.5.1         
[103] blob_1.3.0              prettyunits_1.2.0       latticeExtra_0.6-31    
[106] jpeg_0.1-11             bitops_1.0-9            pwalign_1.4.0          
[109] scales_1.4.0            crayon_1.5.3            GetoptLong_1.1.0       
[112] rlang_1.1.7             KEGGREST_1.48.1        </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
