---
title: "RNA_CorM.Rmd"
output: html_document
date: "2025-11-29"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

```{r package_loading}
library("edgeR")
library("ggplot2")
library("tibble")
library("dplyr")
library("edgeR")
library("ggrepel")
library("readr")
library("org.Hs.eg.db")
library("AnnotationDbi")
library("pheatmap")
library("Cormotif")
library("tidyr")


```


```{r RNA_counts_table}
chimp_RNA_pantro5_fc <- read.delim("~/diff_timeline_tes/RNA/featurecounts/c_samples_counts.txt", comment.char="#",row.names=1)
human_RNA_hg38_fc <- read.delim("~/diff_timeline_tes/RNA/featurecounts/h_sample_counts.txt", comment.char="#",row.names=1)
human_df <- human_RNA_hg38_fc %>%
  tibble::rownames_to_column("gene")
chimp_df <- chimp_RNA_pantro5_fc %>%
  tibble::rownames_to_column("gene")
RNA_joined_fc <- left_join(human_df, chimp_df, by = "gene")

#To keep only OrthoGenes and sample columns
RNA_fc <- RNA_joined_fc[ , !(names(RNA_joined_fc) %in% c("gene","Chr.x","Start.x","End.x","Strand.x","Length.x","Geneid.x","Chr.y","Start.y","End.y","Strand.y","Length.y","Geneid.y"))]
# 89 columns; 7 x 6 = 42 Human Exp, 7 x 6 = 42 Chimp Exp, 4 Human Replicate
rownames(RNA_fc) <- RNA_joined_fc$gene

#Rename column Names to More Useful Info
col_names<- c("H28126_D0",
              "H28126_D2",
              "H28126_D4",
              "H28126_D5",
              "H28126_D15",
              "H28126_D30",
              "H17_D0",
              "H17_D2",
              "H17_D4",
              "H17_D5",
              "H17_D15",
              "H17_D30",
              "H78_D0",
              "H78_D2",
              "H78_D4",
              "H78_D5",
              "H78_D15",
              "H78_D30",
              "H20682_D0",
              "H20682_D2",
              "H20682_D4",
              "H20682_D5",
              "H20682_D15",
              "H20682_D30",
              "H22422_D0",
              "H22422_D2",
              "H22422_D4",
              "H22422_D5",
              "H22422_D15",
              "H22422_D30",
              "H21792_D0",
              "H21792_D2",
              "H21792_D4",
              "H21792_D5",
              "H21792_D15",
              "H21792_D30",
              "H24280_D0",
              "H24280_D2",
              "H24280_D4",
              "H24280_D5",
              "H24280_D15",
              "H24280_D30",
              "H20682R_D0",
              "H20682R_D2",
              "H20682R_D5",
              "H20682R_D30",
              "C3649_D0",
              "C3649_D2",
              "C3649_D4",
              "C3649_D5",
              "C3649_D15",
              "C3649_D30",
              "C4955_D0",
              "C4955_D2",
              "C4955_D4",
              "C4955_D5",
              "C4955_D15",
              "C4955_D30",
              "C3651_D0",
              "C3651_D2",
              "C3651_D4",
              "C3651_D5",
              "C3651_D15",
              "C3651_D30",
              "C40210_D0",
              "C40210_D2",
              "C40210_D4",
              "C40210_D5",
              "C40210_D15",
              "C40210_D30",
              "C8861_D0",
              "C8861_D2",
              "C8861_D4",
              "C8861_D5",
              "C8861_D15",
              "C8861_D30",
              "C40280_D0",
              "C40280_D2",
              "C40280_D4",
              "C40280_D5",
              "C40280_D15",
              "C40280_D30",
              "C3647_D0",
              "C3647_D2",
              "C3647_D4",
              "C3647_D5",
              "C3647_D15",
              "C3647_D30"
)
colnames(RNA_fc)<-col_names
dim(RNA_fc)
```

````{r RNA-Seq_Filtering_NoD4_RowMeans>0}
time_NoD4 <- c(rep(c("D0","D2","D5","D15","D30"),7),"D0","D2","D5","D30",rep(c("D0","D2","D5","D15","D30"),7))
species_NoD4 <- c(rep("H",35),rep("HR",4),rep("C",35))
keep_timepoints_NoD4 <- time_NoD4 %in% c("D0", "D2", "D5", "D15","D30","D0R","D2R","D5R","D30R")

RNA_fc_NoD4 <- RNA_fc[,keep_timepoints_NoD4]
RNA_log2cpm_NoD4 <- cpm(RNA_fc_NoD4,log=TRUE)
row_means_NoD4 <- rowMeans(RNA_log2cpm_NoD4)
Filt_RMG0_RNA_fc_NoD4 <- RNA_fc_NoD4[row_means_NoD4 >0,]
dim(Filt_RMG0_RNA_fc_NoD4)
Filt_RMG0_RNA_log2cpm_NoD4 <- cpm(Filt_RMG0_RNA_fc_NoD4,log=TRUE)
````

````{r RNA-Seq_CorMotif_SetUp}
###### CorMotif Setup (Pooled by Species + Timepoint) ######


#### Step 1: Prepare Data Set
###Remove specific replicates
samples_to_remove_CorM_NoD4_NoRep <- c("H20682R_D0", "H20682R_D2", "H20682_D5", "H20682R_D30",
              "H28126_D4",
              "H17_D4",
              "H78_D4",
              "H20682_D4",
              "H22422_D4",
              "H21792_D4",
              "H24280_D4",
              "C3649_D4",
              "C4955_D4",
              "C3651_D4",
              "C40210_D4",
              "C8861_D4",
              "C40280_D4",
              "C3647_D4")
Filt_RMG0_RNA_fc_NoD4_NoRep <- Filt_RMG0_RNA_fc_NoD4[, !(colnames(Filt_RMG0_RNA_fc_NoD4) %in% samples_to_remove_CorM_NoD4_NoRep)]
dge_NoD4_NoRep <- DGEList(counts=Filt_RMG0_RNA_fc_NoD4_NoRep)
dge_NoD4_NoRep <- calcNormFactors(dge_NoD4_NoRep,method = "TMM")
log2cpm_NoD4_NoRep <- cpm(dge_NoD4_NoRep,log = TRUE, prior.count=1)

# Set up Variables
groupid <- c(rep(1:5,7),rep(6:10,7))
compid_Cond1 <- c(rep(1,4),rep(6,4))
compid_Cond2 <- c(2,3,4,5,7,8,9,10)
compid <- cbind(compid_Cond1,compid_Cond2)

#Model Fitting
set.seed(234)
#motif.fitted<-cormotiffit(log2cpm_NoD4_NoRep,groupid,compid,K=(10),max.iter=2000,BIC=TRUE)
#motif.fitted$bic
head(motif.fitted$bestmotif$p.post)
dif.pattern.RNA<-(motif.fitted$bestmotif$p.post>0.5)
head(dif.pattern.RNA)
tail(dif.pattern.RNA)
topgenelist<-generank(motif.fitted$bestmotif$p.post)
motif_assignments <- apply(motif.fitted$bestmotif$p.post, 1, which.max)
table(motif_assignments)
````
```{r PCA RNA-Seq RowMeans Filtering ASHG Subset,fig.height=5,fig.width=9}

plotMotif(motif.fitted)

```

```{r Manuel Plotting}

# Extract motif matrix
motif_matrix <- as.matrix(motif.fitted$bestmotif$motif.q)

# Label rows and columns
rownames(motif_matrix) <- paste0("Motif_", seq_len(nrow(motif_matrix)))
colnames(motif_matrix) <- paste0("C", compid[,1], "_vs_C", compid[,2])

# Convert to tidy format
motif_df <- as.data.frame(motif_matrix) %>%
  mutate(Motif = rownames(.)) %>%
  pivot_longer(-Motif, names_to = "Comparison", values_to = "Probability")

# Reorder motifs 1–11
motif_df$Motif <- factor(motif_df$Motif, levels = paste0("Motif_", 1:10))

# Plot
ggplot(motif_df, aes(x = Comparison, y = Motif, fill = Probability)) +
  geom_tile(color = "white") +
  scale_fill_gradient(low = "white", high = "black") +
  labs(
    title = "CorMotif Motif Activity Across Comparisons \n  (no day4) (no replicates) (RowMeans<0)",
    x = "Comparison (Condition 1 vs Condition 2)",
    y = "Motif (1–10)",
    fill = "Posterior Probability"
  ) +
  theme_minimal(base_size = 14) +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    panel.grid = element_blank()
  )


```
