---
title: "QC_Checks"
output: html_document
date: "2025-11-29"
---
```{r package loading}
library("edgeR")
library("ggplot2")
library("tibble")
library("dplyr")
library("edgeR")
library("ggrepel")
library("readr")
library("org.Hs.eg.db")
library("AnnotationDbi")
library("pheatmap")
library("Cormotif")
```

```{r H3J27ac counts table}
chimp_RNA_pantro5_fc <- read.delim("~/diff_timeline_tes/RNA/featurecounts/c_samples_counts.txt", comment.char="#",row.names=1)
human_RNA_hg38_fc <- read.delim("~/diff_timeline_tes/RNA/featurecounts/h_sample_counts.txt", comment.char="#",row.names=1)
human_df <- human_RNA_hg38_fc %>%
  tibble::rownames_to_column("gene")
chimp_df <- chimp_RNA_pantro5_fc %>%
  tibble::rownames_to_column("gene")
RNA_joined_fc <- left_join(human_df, chimp_df, by = "gene")

#To keep only OrthoGenes and sample columns
RNA_fc <- RNA_joined_fc[ , !(names(RNA_joined_fc) %in% c("gene","Chr.x","Start.x","End.x","Strand.x","Length.x","Geneid.x","Chr.y","Start.y","End.y","Strand.y","Length.y","Geneid.y"))]
# 89 columns; 7 x 6 = 42 Human Exp, 7 x 6 = 42 Chimp Exp, 4 Human Replicate
rownames(RNA_fc) <- RNA_joined_fc$gene

#Rename column Names to More Useful Info
col_names<- c("H28126_D0",
              "H28126_D2",
              "H28126_D4",
              "H28126_D5",
              "H28126_D15",
              "H28126_D30",
              "H17_D0",
              "H17_D2",
              "H17_D4",
              "H17_D5",
              "H17_D15",
              "H17_D30",
              "H78_D0",
              "H78_D2",
              "H78_D4",
              "H78_D5",
              "H78_D15",
              "H78_D30",
              "H20682_D0",
              "H20682_D2",
              "H20682_D4",
              "H20682_D5",
              "H20682_D15",
              "H20682_D30",
              "H22422_D0",
              "H22422_D2",
              "H22422_D4",
              "H22422_D5",
              "H22422_D15",
              "H22422_D30",
              "H21792_D0",
              "H21792_D2",
              "H21792_D4",
              "H21792_D5",
              "H21792_D15",
              "H21792_D30",
              "H24280_D0",
              "H24280_D2",
              "H24280_D4",
              "H24280_D5",
              "H24280_D15",
              "H24280_D30",
              "H20682R_D0",
              "H20682R_D2",
              "H20682R_D5",
              "H20682R_D30",
              "C3649_D0",
              "C3649_D2",
              "C3649_D4",
              "C3649_D5",
              "C3649_D15",
              "C3649_D30",
              "C4955_D0",
              "C4955_D2",
              "C4955_D4",
              "C4955_D5",
              "C4955_D15",
              "C4955_D30",
              "C3651_D0",
              "C3651_D2",
              "C3651_D4",
              "C3651_D5",
              "C3651_D15",
              "C3651_D30",
              "C40210_D0",
              "C40210_D2",
              "C40210_D4",
              "C40210_D5",
              "C40210_D15",
              "C40210_D30",
              "C8861_D0",
              "C8861_D2",
              "C8861_D4",
              "C8861_D5",
              "C8861_D15",
              "C8861_D30",
              "C40280_D0",
              "C40280_D2",
              "C40280_D4",
              "C40280_D5",
              "C40280_D15",
              "C40280_D30",
              "C3647_D0",
              "C3647_D2",
              "C3647_D4",
              "C3647_D5",
              "C3647_D15",
              "C3647_D30"
)
colnames(RNA_fc)<-col_names
dim(RNA_fc)
```
```{r - Boxplot and histograms}
#####Unfiltered####
RNA_log2cpm <- cpm(RNA_fc,log=TRUE)
hist(RNA_log2cpm,  main = "Histogram of all counts (unfiltered)",
     xlab =expression("Log"[2]*" counts-per-million"), col =4 )
boxplot(RNA_log2cpm, main = "Boxplots of log cpm per sample (unfiltered)",xaxt = "n", xlab= "")
axis(1,
     at = 1:length(col_names),  # positions (one per sample)
     labels = col_names,        # your labels vector
     las = 2,               # rotate text vertically (like srt=90)
     cex.axis = 0.3)        # shrink label size


timepoint <- c(rep(c("D0","D2","D4","D5","D15","D30"),7),"D0R","D2R","D5R","D30R", rep(c("D0","D2","D4","D5","D15","D30"),7))
time_order <- c("D0", "D2", "D4", "D5", "D15", "D30")
species <- c(rep("H",42),rep("HR",4),rep("C",42))

T_RNA_log2cpm <- t(RNA_log2cpm)
pca_raw <- prcomp(T_RNA_log2cpm,scale=TRUE,center=TRUE)
pca_raw_meta <- as.data.frame(pca_raw$x)
pca_raw_meta$sample <- rownames(pca_raw_meta)
pca_raw_meta$species <- species
pca_raw_meta$timepoint <- timepoint
pca_raw_meta$timepoint <- factor(pca_raw_meta$timepoint, levels = time_order)
dim(T_RNA_log2cpm)
```


```{r PCA RNA-Seq Unfiltered,fig.height=4,fig.width=9}
ggplot(pca_raw_meta, aes(x = PC1, y = PC2, color = timepoint, shape = species)) +
  geom_point(size = 3, alpha = 0.8) +
  geom_text_repel(aes(label = sample), size = 3, max.overlaps = 100) +
  theme_minimal() +
  labs(title = "PCA of log2CPM RNA-Seq Data Unfiltered",
       x = paste0("PC1 (", round(summary(pca_raw)$importance[2, 1] * 100, 1), "%)"),
       y = paste0("PC2 (", round(summary(pca_raw)$importance[2, 2] * 100, 1), "%)"))

```

```{r Filtering based on Row Means}
#####RowMu>0####
row_means <- rowMeans(RNA_log2cpm)
Filt_RMG0_RNA_fc <- RNA_fc[row_means >0,]
Filt_RMG0_RNA_log2cpm <- cpm(Filt_RMG0_RNA_fc,log=TRUE)

hist(Filt_RMG0_RNA_log2cpm, main = "Histogram of filtered counts using rowMeans > 0 method",
     xlab =expression("Log"[2]*" counts-per-million"), col =5 )
boxplot(Filt_RMG0_RNA_log2cpm, main = "Boxplots of log cpm per sample (RowMeans>0)",xaxt = "n", xlab= "")
axis(1,
     at = 1:length(col_names),  # positions (one per sample)
     labels = col_names,        # your labels vector
     las = 2,               # rotate text vertically (like srt=90)
     cex.axis = 0.3)        # shrink label size
```


```{r PCA RNA-Seq RowMeans Filtering,fig.height=4,fig.width=9}
######PCA####
T_Filt_RMG0_RNA_log2cpm <- t(Filt_RMG0_RNA_log2cpm)
pca_RMG0 <- prcomp(T_Filt_RMG0_RNA_log2cpm,scale=TRUE,center=TRUE)
pca_RMG0_meta <- as.data.frame(pca_RMG0$x)
pca_RMG0_meta$sample <- rownames(pca_RMG0_meta)
pca_RMG0_meta$species <- species
pca_RMG0_meta$timepoint <- timepoint
pca_RMG0_meta$timepoint <- factor(pca_RMG0_meta$timepoint, levels = time_order)
dim(T_Filt_RMG0_RNA_log2cpm)

#PC1 vs PC2
ggplot(pca_RMG0_meta, aes(x = PC1, y = PC2, color = timepoint, shape = species)) +
  geom_point(size = 3, alpha = 0.8) +
  geom_text_repel(aes(label = sample), size = 3, max.overlaps = 100) +
  theme_minimal() +
  labs(title = "PCA of log2CPM RNA-Seq Data (RowMean>0)",
       x = paste0("PC1 (", round(summary(pca_RMG0)$importance[2, 1] * 100, 1), "%)"),
       y = paste0("PC2 (", round(summary(pca_RMG0)$importance[2, 2] * 100, 1), "%)"))

######Cor_HeatMap####
cor_Filt_RMG0_RNA_log2cpm <- cor(Filt_RMG0_RNA_log2cpm, method = "pearson")
time <- factor(timepoint,levels = c("D0","D2","D4","D5","D15","D30","D0R","D2R","D5R","D30R"))
metadata <- data.frame(
  sample_cor = colnames(Filt_RMG0_RNA_log2cpm),
  species_cor = species,
  timepoint_cor = timepoint
)
ann_colors <- list(
  timepoint_cor = c(
    "D0" = "#883268",   # Purple
    "D2" = "#3E7274",  # blue
    "D4" = "#5AAA464D",  # light green
    "D5" = "#94C47D",  # Green
    "D15" = "#C03830",  # red
    "D30" = "#830C05",  # dark red
    "D0R" = "#883268",  # pink
    "D2R" = "#3E7274",  # teal
    "D5R" = "#94C47D",  # bright green
    "D30R" = "#830C05"  # bright red
  ),
  species_cor = c(
    "H" = "#171717",  # black
    "C" = "#17171717",   # light grey
    "HR" = "#171717B0"#Dark Grey
  )
)
rownames(metadata) <- metadata$sample_cor
```


```{r Correlation HeatMap RNA-Seq RowMeans Filtering,fig.height=10,fig.width=10}
pheatmap(cor_Filt_RMG0_RNA_log2cpm,
         fontsize_row = 5,
         fontsize_col = 5,
         annotation_col = metadata[, c("species_cor", "timepoint_cor")],
         annotation_colors = ann_colors,
         clustering_distance_rows = "correlation",
         clustering_distance_cols = "correlation",
         main = "Sample-Sample Correlation (Pearson-log2CPM-RowMeans>0)")

```

```{r Onmitting Day 4 of Data}
#####Subset=ASHG####
time_ASHG <- c(rep(c("D0","D2","D5","D15","D30"),7),"D0","D2","D5","D30",rep(c("D0","D2","D5","D15","D30"),7))
species_ASHG <- c(rep("H",35),rep("HR",4),rep("C",35))
keep_timepoints <- time %in% c("D0", "D2", "D5", "D15","D30","D0R","D2R","D5R","D30R")

RNA_fc_ASHGSubset <- RNA_fc[,keep_timepoints]
RNA_log2cpm_ASHGSubset <- cpm(RNA_fc_ASHGSubset,log=TRUE)
row_means_ASHGSubset <- rowMeans(RNA_log2cpm_ASHGSubset)
Filt_RMG0_RNA_fc_ASHGSubset <- RNA_fc_ASHGSubset[row_means_ASHGSubset >0,]
dim(Filt_RMG0_RNA_fc_ASHGSubset)
Filt_RMG0_RNA_log2cpm_ASHGSubset <- cpm(Filt_RMG0_RNA_fc_ASHGSubset,log=TRUE)

T_Filt_RMG0_RNA_log2cpm_ASHGSubset <- t(Filt_RMG0_RNA_log2cpm_ASHGSubset)
pca_ASHG <- prcomp(T_Filt_RMG0_RNA_log2cpm_ASHGSubset,scale=TRUE,center=TRUE)
pca_ASHG_meta <- as.data.frame(pca_ASHG$x)
pca_ASHG_meta$sample <- rownames(pca_ASHG_meta)
pca_ASHG_meta$species <- species_ASHG
pca_ASHG_meta$timepoint <- time_ASHG
time_ASHG_level <- as.factor(c("D0","D2","D5","D15","D30"))
pca_ASHG_meta$timepoint <- factor(pca_ASHG_meta$timepoint, levels = time_ASHG_level)
dim(T_Filt_RMG0_RNA_log2cpm_ASHGSubset)
```


```{r PCA RNA-Seq RowMeans Filtering ASHG Subset,fig.height=4,fig.width=9}
ggplot(pca_ASHG_meta, aes(x = PC1, y = PC2, color = timepoint, shape = species)) +
  geom_point(size = 3, alpha = 0.8) +
  geom_text_repel(aes(label = sample), size = 3, max.overlaps = 100) +
  theme_minimal() +
  labs(title = "PCA of log2CPM (D0, D2, D5, and D15) (rowmeans >0)",
       x = paste0("PC1 (", round(summary(pca_ASHG)$importance[2, 1] * 100, 1), "%)"),
       y = paste0("PC2 (", round(summary(pca_ASHG)$importance[2, 2] * 100, 1), "%)"))


```


```{r CorMotif}
###### CorMotif Setup (Pooled by Species + Timepoint) ######


#### Step 1: Prepare Data Set
###Remove specific replicates
samples_to_remove_CorM <- c("H20682R_D0", "H20682R_D2", "H20682_D5", "H20682R_D30",
              "H28126_D4",
              "H17_D4",
              "H78_D4",
              "H20682_D4",
              "H22422_D4",
              "H21792_D4",
              "H24280_D4",
              "C3649_D4",
              "C4955_D4",
              "C3651_D4",
              "C40210_D4",
              "C8861_D4",
              "C40280_D4",
              "C3647_D4")
Filt_RMG0_RNA_fc_NoRep <- Filt_RMG0_RNA_fc[, !(colnames(Filt_RMG0_RNA_fc) %in% samples_to_remove_CorM)]
dge <- DGEList(counts=Filt_RMG0_RNA_fc_NoRep)
dge <- calcNormFactors(dge,method = "TMM")
log2cpm <- cpm(dge,log = TRUE, prior.count=1)

# Set up Variables
groupid <- c(rep(1:5,7),rep(6:10,7))
compid_Cond1 <- c(rep(1,4),rep(6,4))
compid_Cond2 <- c(2,3,4,5,7,8,9,10)
compid <- cbind(compid_Cond1,compid_Cond2)

#Model Fitting
set.seed(234)
motif.fitted<-cormotiffit(log2cpm,groupid,compid,K=(10),max.iter=2000,BIC=TRUE)
#motif.fitted$bic
plotMotif(motif.fitted)
head(motif.fitted$bestmotif$p.post)
dif.pattern.RNA<-(motif.fitted$bestmotif$p.post>0.5)
head(dif.pattern.RNA)
tail(dif.pattern.RNA)
topgenelist<-generank(motif.fitted$bestmotif$p.post)
motif_assignments <- apply(motif.fitted$bestmotif$p.post, 1, which.max)
table(motif_assignments)

#Model Fitting
#set.seed(91011)
#motif.fitted<-cormotiffit(log2cpm,groupid,compid,K=(10),max.iter=2000,BIC=TRUE)
#motif.fitted$bic
#plotIC(motif.fitted)
#plotMotif(motif.fitted)
#head(motif.fitted$bestmotif$p.post)
#dif.pattern.RNA<-(motif.fitted$bestmotif$p.post>0.5)
#head(dif.pattern.RNA)
#tail(dif.pattern.RNA)
#topgenelist<-generank(motif.fitted$bestmotif$p.post)
#motif_assignments <- apply(motif.fitted$bestmotif$p.post, 1, which.max)
#table(motif_assignments)

```
```{r PCA RNA-Seq RowMeans Filtering ASHG Subset,fig.height=5,fig.width=9}

plotMotif(motif.fitted)

```

```{r Manuel Plotting}
library(ggplot2)
library(tidyr)
library(dplyr)

# Extract motif matrix
motif_matrix <- as.matrix(motif.fitted$bestmotif$motif.q)

# Label rows and columns
rownames(motif_matrix) <- paste0("Motif_", seq_len(nrow(motif_matrix)))
colnames(motif_matrix) <- paste0("C", compid[,1], "_vs_C", compid[,2])

# Convert to tidy format
motif_df <- as.data.frame(motif_matrix) %>%
  mutate(Motif = rownames(.)) %>%
  pivot_longer(-Motif, names_to = "Comparison", values_to = "Probability")

# Reorder motifs 1–11
motif_df$Motif <- factor(motif_df$Motif, levels = paste0("Motif_", 1:10))

# Plot
ggplot(motif_df, aes(x = Comparison, y = Motif, fill = Probability)) +
  geom_tile(color = "white") +
  scale_fill_gradient(low = "white", high = "black") +
  labs(
    title = "CorMotif Motif Activity Across Comparisons",
    x = "Comparison (Condition 1 vs Condition 2)",
    y = "Motif (1–10)",
    fill = "Posterior Probability"
  ) +
  theme_minimal(base_size = 14) +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    panel.grid = element_blank()
  )


```

